--- third_party/blink/renderer/platform/widget/compositing/layer_tree_settings.cc
+++ third_party/blink/renderer/platform/widget/compositing/layer_tree_settings.cc
@@ -182,18 +182,10 @@
     actual.bytes_limit_when_visible = 256 * 1024 * 1024;
   }
 #else
-<<<<<<< HEAD
   if (base::FeatureList::IsEnabled(kIncreaseTileMemorySizeProportionally)) {
     // This calculation will increase the tile memory size. It should apply to
     // the other plateforms if no regression on Mac.
-    actual.priority_cutoff_when_visible =
-        gpu::MemoryAllocation::CUTOFF_ALLOW_NICE_TO_HAVE;
-=======
-  // Ignore what the system said and give all clients the same maximum
-  // allocation on desktop platforms.
-  actual.bytes_limit_when_visible = 512 * 1024 * 1024;
->>>>>>> d9b50f89921 ([cc] Use simple compositor memory limits on Android (#4637))
-
+    //
     // For large monitors with high resolution, increase the tile memory to
     // avoid frequent out of memory problems. With Mac M1 on
     // https://www.334-28th.com/, it seems 512 MB works fine on 1920x1080 * 2
@@ -226,8 +218,6 @@
     // Ignore what the system said and give all clients the same maximum
     // allocation on desktop platforms.
     actual.bytes_limit_when_visible = 512 * 1024 * 1024;
-    actual.priority_cutoff_when_visible =
-        gpu::MemoryAllocation::CUTOFF_ALLOW_NICE_TO_HAVE;
 
     // For large monitors (4k), double the tile memory to avoid frequent out of
     // memory problems. 4k could mean a screen width of anywhere from 3840 to
@@ -241,11 +231,8 @@
     }
   }
 #endif
-<<<<<<< HEAD
-=======
   actual.priority_cutoff_when_visible =
       gpu::MemoryAllocation::CUTOFF_ALLOW_NICE_TO_HAVE;
->>>>>>> d9b50f89921 ([cc] Use simple compositor memory limits on Android (#4637))
 
   return actual;
 }
