--- media/base/decoder_buffer.h
+++ media/base/decoder_buffer.h
@@ -296,16 +296,10 @@
   // Returns a human-readable string describing |*this|.
   std::string AsHumanReadableString(bool verbose = false) const;
 
-<<<<<<< HEAD
-=======
-  // Replaces any existing side data with data copied from |side_data|.
-  void CopySideDataFrom(const uint8_t* side_data, size_t side_data_size);
-
   // Returns total memory usage for both bookkeeping and buffered data. The
   // function is added for more accurately memory management.
   virtual size_t GetMemoryUsage() const;
 
->>>>>>> eb9daec563f9 (Refine Demuxer GC for low resolution videos (#4483))
  protected:
   friend class base::RefCountedThreadSafe<DecoderBuffer>;
 
