--- base/time/time_unittest.cc
+++ base/time/time_unittest.cc
@@ -28,16 +28,9 @@
 #include "third_party/icu/source/i18n/unicode/timezone.h"
 
 #if BUILDFLAG(IS_STARBOARD)
-<<<<<<< HEAD
-#include "base/logging.h"
-#include "starboard/common/time.h"  // nogncheck
-#include "starboard/types.h"  // nogncheck
-#include "starboard/client_porting/eztime/eztime.h"  // nogncheck
-=======
 #include "cobalt/common/eztime/eztime.h"  // nogncheck
 #include "starboard/common/time.h"        // nogncheck
 #include "starboard/types.h"              // nogncheck
->>>>>>> 144a6739d60 (Move eztime out of Starboard (#5765))
 #elif BUILDFLAG(IS_ANDROID)
 #include "base/android/jni_android.h"
 #elif BUILDFLAG(IS_FUCHSIA) || BUILDFLAG(IS_CHROMEOS)
