--- content/web_test/browser/web_test_content_browser_client.h
+++ content/web_test/browser/web_test_content_browser_client.h
@@ -176,16 +176,13 @@
       int render_process_id,
       mojo::PendingAssociatedReceiver<mojom::WebTestControlHost> receiver);
 
-<<<<<<< HEAD
   void BindNonAssociatedWebTestControlHost(
       mojo::PendingReceiver<mojom::NonAssociatedWebTestControlHost> receiver);
-=======
 #if BUILDFLAG(IS_COBALT)
   void BindH5vccRuntime(
       RenderFrameHost* render_frame_host,
       mojo::PendingReceiver<h5vcc_runtime::mojom::H5vccRuntime> receiver);
 #endif
->>>>>>> 44abbc07f1f (Add binder for H5vccRuntime to fix other web tests (#5183))
 
   bool block_popups_ = true;
   bool screen_orientation_changed_ = false;
@@ -198,17 +195,12 @@
   std::unique_ptr<WebTestSensorProviderManager> sensor_provider_manager_;
   mojo::UniqueReceiverSet<blink::test::mojom::CookieManagerAutomation>
       cookie_managers_;
-<<<<<<< HEAD
   mojo::UniqueReceiverSet<blink::test::mojom::FederatedAuthRequestAutomation>
       fedcm_managers_;
-=======
-  mojo::UniqueReceiverSet<blink::test::mojom::AttributionReportingAutomation>
-      attribution_reporting_receivers_;
 
 #if BUILDFLAG(IS_COBALT)
   std::unique_ptr<StubH5vccRuntimeImpl> stub_h5vcc_runtime_impl_;
 #endif
->>>>>>> 44abbc07f1f (Add binder for H5vccRuntime to fix other web tests (#5183))
 };
 
 }  // namespace content
