--- content/web_test/browser/web_test_content_browser_client.cc
+++ content/web_test/browser/web_test_content_browser_client.cc
@@ -550,23 +550,16 @@
   map->Add<blink::test::mojom::CookieManagerAutomation>(base::BindRepeating(
       &WebTestContentBrowserClient::BindCookieManagerAutomation,
       base::Unretained(this)));
-<<<<<<< HEAD
   map->Add<blink::test::mojom::FederatedAuthRequestAutomation>(
       base::BindRepeating(&WebTestContentBrowserClient::BindFedCmAutomation,
                           base::Unretained(this)));
   map->Add<blink::test::mojom::WebSensorProviderAutomation>(base::BindRepeating(
       &WebTestContentBrowserClient::BindWebSensorProviderAutomation,
       base::Unretained(this)));
-=======
-  map->Add<blink::test::mojom::AttributionReportingAutomation>(
-      base::BindRepeating(
-          &WebTestContentBrowserClient::BindAttributionReportingAutomation,
-          base::Unretained(this)));
 #if BUILDFLAG(IS_COBALT)
   map->Add<h5vcc_runtime::mojom::H5vccRuntime>(base::BindRepeating(
       &WebTestContentBrowserClient::BindH5vccRuntime, base::Unretained(this)));
 #endif
->>>>>>> 44abbc07f1f (Add binder for H5vccRuntime to fix other web tests (#5183))
 }
 
 bool WebTestContentBrowserClient::CanAcceptUntrustedExchangesIfNeeded() {
