--- third_party/blink/renderer/core/workers/cross_thread_global_scope_creation_params_copier.cc
+++ third_party/blink/renderer/core/workers/cross_thread_global_scope_creation_params_copier.cc
@@ -67,15 +67,12 @@
       in->allow_self, in->allow_star, in->allow_inline,
       in->allow_inline_speculation_rules, in->allow_eval, in->allow_wasm_eval,
       in->allow_wasm_unsafe_eval, in->allow_dynamic, in->allow_unsafe_hashes,
-<<<<<<< HEAD
-      in->report_sample, in->report_hash_algorithm);
-=======
 #if BUILDFLAG(IS_COBALT)
-      in->report_sample, in->cobalt_insecure_local_network);
+      in->report_sample, in->report_hash_algorithm,
+      in->cobalt_insecure_local_network);
 #else
-      in->report_sample);
+      in->report_sample, in->report_hash_algorithm);
 #endif
->>>>>>> ec4e2c2ce45 (Add custom cobalt-insecure-local-network csp source (#4958))
 }
 
 HashMap<network::mojom::blink::CSPDirectiveName,
