--- build/build_config.h
+++ build/build_config.h
@@ -17,15 +17,9 @@
 //  Operating System:
 //    IS_AIX / IS_ANDROID / IS_ASMJS / IS_CHROMEOS / IS_FREEBSD / IS_FUCHSIA /
 //    IS_IOS / IS_IOS_MACCATALYST / IS_IOS_TVOS / IS_LINUX / IS_MAC / IS_NACL /
-<<<<<<< HEAD
 //    IS_NETBSD / IS_OPENBSD / IS_QNX / IS_SOLARIS / IS_WATCHOS / IS_WIN
 //  Operating System family:
 //    IS_APPLE: IOS or MAC or IOS_MACCATALYST or IOS_TVOS or WATCHOS
-=======
-//    IS_NETBSD / IS_OPENBSD / IS_QNX / IS_SOLARIS / IS_WIN
-//  Operating System family:
-//    IS_APPLE: IOS or MAC or IOS_MACCATALYST or IOS_TVOS
->>>>>>> e6e7e9c54f5 ([tvOS] Add minimum set of required backports + tvos-arm64-simulator build config (#7107))
 //    IS_BSD: FREEBSD or NETBSD or OPENBSD
 //    IS_POSIX: AIX or ANDROID or ASMJS or CHROMEOS or FREEBSD or IOS or LINUX
 //              or MAC or NACL or NETBSD or OPENBSD or QNX or SOLARIS
@@ -86,12 +80,9 @@
 #if defined(TARGET_OS_TV) && TARGET_OS_TV
 #define OS_IOS_TVOS 1
 #endif  // defined(TARGET_OS_TV) && TARGET_OS_TV
-<<<<<<< HEAD
 #if defined(TARGET_OS_WATCH) && TARGET_OS_WATCH
 #define OS_WATCHOS 1
 #endif  // defined(TARGET_OS_WATCH) && TARGET_OS_WATCH
-=======
->>>>>>> e6e7e9c54f5 ([tvOS] Add minimum set of required backports + tvos-arm64-simulator build config (#7107))
 #else
 #define OS_MAC 1
 #endif  // defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
