--- content/renderer/media/media_factory.cc
+++ content/renderer/media/media_factory.cc
@@ -911,14 +911,10 @@
 #if BUILDFLAG(IS_FUCHSIA)
   DCHECK(interface_broker_);
   cdm_factory_ = std::make_unique<media::FuchsiaCdmFactory>(
-<<<<<<< HEAD
       std::make_unique<media::MojoFuchsiaCdmProvider>(interface_broker_),
       GetKeySystems());
-=======
-      std::make_unique<media::MojoFuchsiaCdmProvider>(interface_broker_));
 #elif BUILDFLAG(USE_STARBOARD_MEDIA)
   cdm_factory_ = std::make_unique<media::StarboardCdmFactory>();
->>>>>>> 6d979993fcb8 (Implement media::ContentDecryptionModule interface for Starboard (#4368))
 #elif BUILDFLAG(ENABLE_MOJO_CDM)
   cdm_factory_ = std::make_unique<media::MojoCdmFactory>(
       GetMediaInterfaceFactory(), GetKeySystems());
