--- BUILD.gn
+++ BUILD.gn
@@ -871,7 +871,6 @@
       deps += [ "//components/cronet:cronet_tests" ]
     }
   }
-<<<<<<< HEAD
 
   if (is_chrome_for_testing) {
     deps += [ "//chrome/browser/chrome_for_testing:chrome_for_testing" ]
@@ -880,8 +879,9 @@
   if (checkout_src_internal) {
     deps += [ "//internal:all" ]
   }
-=======
->>>>>>> b9e351ee8f5 (Revert "Revert "Build starboard shared library for linux-x64x11 (#4215)"" (#4247))
+  if (chrobalt) {
+    deps += [ "//chrobalt:gn_all" ]
+  }
   if (chrobalt) {
     deps += [ "//chrobalt:gn_all" ]
   }
