--- cobalt/build/configs/common.gn
+++ cobalt/build/configs/common.gn
@@ -49,10 +49,8 @@
 # This will force V8 to embed snapshot blob in binary.
 v8_use_external_startup_data = false
 
-<<<<<<< HEAD
 # Disable hls_demuxer so that enable_mse_mpeg2ts_stream_parser=false
 enable_hls_demuxer=false
-=======
+
 # Enable widevine for encrypted media playback.
 enable_widevine = true
->>>>>>> 8b8011690b5 (Enable support for widevine  (#7248))
