--- cobalt/build/configs/common.gn
+++ cobalt/build/configs/common.gn
@@ -47,10 +47,8 @@
 # This flag is also needed to disable concurrent marking.
 v8_enable_atomic_object_field_writes = false
 
-<<<<<<< HEAD
-# Disable hls_demuxer so that enable_mse_mpeg2ts_stream_parser=false
-enable_hls_demuxer=false
-=======
 # This will force V8 to embed snapshot blob in binary.
 v8_use_external_startup_data = false
->>>>>>> f3b6f212eb1 (Use embedded v8 snapshot for linux (#6426))
+
+# Disable hls_demuxer so that enable_mse_mpeg2ts_stream_parser=false
+enable_hls_demuxer=false
