--- content/public/renderer/content_renderer_client.h
+++ content/public/renderer/content_renderer_client.h
@@ -456,21 +456,19 @@
   CreateCastStreamingResourceProvider();
 #endif
 
-<<<<<<< HEAD
   // Creates a WebLinkPreviewTriggerer if an embedder wants to observe events
   // and trigger preview. It is allowed to return nullptr.
   //
   // See blink::WebLinkPreviewTriggerer for more details.
   virtual std::unique_ptr<blink::WebLinkPreviewTriggerer>
   CreateLinkPreviewTriggerer();
-=======
+
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
   // This is used for StarboardRenderer to acquire customizations
   // as MojoRenderer from cobalt_content_renderer_client.cc.
   virtual void GetStarboardRendererFactoryTraits(
       media::RendererFactoryTraits* renderer_factory_traits);
 #endif // BUILDFLAG(USE_STARBOARD_MEDIA)
->>>>>>> 73a1238456a ([media] Move StarboardRenderer to MojoRenderer (#5113))
 };
 
 }  // namespace content
