--- ui/gl/gl_features.cc
+++ ui/gl/gl_features.cc
@@ -217,7 +217,6 @@
           kPassthroughCommandDecoderBlockListByAndroidBuildFP.Get()))
     return false;
 
-<<<<<<< HEAD
   // Only check system info once and cache if the vendor is blocked.
   static std::optional<bool> gpu_vendor_blocked;
   if (!gpu_vendor_blocked.has_value()) {
@@ -241,14 +240,13 @@
   if (gpu_vendor_blocked.value()) {
     return false;
   }
-=======
+
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
   // TODO(crbug.com/41482978): Fix single process rendering on nvidia Shield.
   if (base::MatchPattern(build_info->manufacturer(), "NVIDIA") &&
       base::MatchPattern(build_info->model(), "SHIELD*"))
     return false;
 #endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
->>>>>>> e86b5ca0549 (Limit rendering restrictions to shield (#4919))
 #endif  // BUILDFLAG(IS_ANDROID)
 
   return true;
