--- content/browser/browser_child_process_host_impl.cc
+++ content/browser/browser_child_process_host_impl.cc
@@ -393,13 +393,8 @@
   if (!child_process_launcher_) {
     // If the delegate doesn't use Launch() helper.
     ChildProcessTerminationInfo info;
-<<<<<<< HEAD
     // TODO(crbug.com/40255458): iOS is single process mode for now.
-#if !BUILDFLAG(IS_IOS)
-=======
-    // TODO(crbug.com/1412835): iOS is single process mode for now.
 #if !BUILDFLAG(IS_IOS) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
->>>>>>> e153d47e7e6 (Remove zygote processes (#5680))
     info.status = base::GetTerminationStatus(data_.GetProcess().Handle(),
                                              &info.exit_code);
 #endif
