--- content/shell/renderer/shell_content_renderer_client.cc
+++ content/shell/renderer/shell_content_renderer_client.cc
@@ -54,7 +54,6 @@
 #include "media/base/media_switches.h"
 #endif
 
-<<<<<<< HEAD
 #if (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)) && \
     (defined(ARCH_CPU_X86_64) || defined(ARCH_CPU_ARM64))
 #define ENABLE_WEB_ASSEMBLY_TRAP_HANDLER_LINUX
@@ -62,9 +61,6 @@
 #include "v8/include/v8-wasm-trap-handler-posix.h"
 #endif
 
-#if BUILDFLAG(IS_COBALT)
-=======
->>>>>>> c13213fb8c5a (Clean up existing media buildflags (#4393))
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
 #include "starboard/media.h"
 #endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
