--- base/message_loop/message_pump_for_ui.h
+++ base/message_loop/message_pump_for_ui.h
@@ -34,14 +34,9 @@
 // Windows defines it as-is.
 using MessagePumpForUI = MessagePumpForUI;
 #elif BUILDFLAG(IS_ANDROID)
-<<<<<<< HEAD
 using MessagePumpForUI = MessagePumpAndroid;
-=======
-// Android defines it as-is.
-using MessagePumpForUI = MessagePumpForUI;
 #elif BUILDFLAG(IS_STARBOARD)
 using MessagePumpForUI = MessagePumpUIStarboard;
->>>>>>> 9ff7164ebbb (Use the Starboard message pump. (#4772))
 #elif BUILDFLAG(IS_APPLE)
 // MessagePumpForUI isn't bound to a specific impl on Mac. While each impl can
 // be represented by a plain MessagePump: message_pump_apple::Create() must be
