--- third_party/blink/renderer/core/dom/events/event_target.cc
+++ third_party/blink/renderer/core/dom/events/event_target.cc
@@ -1030,16 +1030,10 @@
   return GetDispatchEventResult(event);
 }
 
-<<<<<<< HEAD
 // Fire event listeners, creates a copy of EventListenerVector on being called.
 bool EventTarget::FireEventListeners(Event& event,
                                      EventTargetData* d,
                                      EventListenerVectorSnapshot entry) {
-=======
-bool EventTarget::FireEventListeners(Event& event,
-                                     EventTargetData* d,
-                                     EventListenerVector& entry) {
->>>>>>> d1c2e13a850 (Revert "BACKPORT: Improve event listener dispatch" (#6398))
   // Fire all listeners registered for this event. Don't fire listeners removed
   // during event dispatch. Also, don't fire event listeners added during event
   // dispatch. Conveniently, all new event listeners will be added after or at
