diff --git a/cobalt/build/configs/evergreen-arm-hardfp-rdk/args.gn b/cobalt/build/configs/evergreen-arm-hardfp-rdk/args.gn
index 4a09baf8c2d6..85b05a793810 100644
--- a/cobalt/build/configs/evergreen-arm-hardfp-rdk/args.gn
+++ b/cobalt/build/configs/evergreen-arm-hardfp-rdk/args.gn
@@ -8,3 +8,4 @@ use_evergreen = true
 rdk_build_with_yocto = false
 rdk_starboard_root = "//starboard/contrib/rdk/src"
 rdk_enable_securityagent = false
+starboard_level_final_executable_type = "shared_library"
diff --git a/starboard/contrib/rdk/src/third_party/starboard/rdk/shared/BUILD.gn b/starboard/contrib/rdk/src/third_party/starboard/rdk/shared/BUILD.gn
index 09841217ec5d..76d10b37f938 100644
--- a/starboard/contrib/rdk/src/third_party/starboard/rdk/shared/BUILD.gn
+++ b/starboard/contrib/rdk/src/third_party/starboard/rdk/shared/BUILD.gn
@@ -386,3 +386,33 @@ copy("rdk_cobalt_keymap") {
   sources = [ "cobalt/content/etc/keymapping.json" ]
   outputs = [ "$root_out_dir/app/cobalt/content/etc/keymapping.json" ]
 }
+
+# These changes are temporary for backward compatibility.
+# TODO: b/499089012 - Remove this group once we fully transition to the new
+# plugin method (libloader_app.so) as part of b/491573728.
+group("loader_app_rdk_executable") {
+  data_deps = [":loader_app($starboard_toolchain)"]
+}
+
+if (sb_is_evergreen_compatible && current_toolchain == starboard_toolchain) {
+# These changes are temporary for backward compatibility.
+# TODO: b/499089012 - Remove once we move to the new plugin method (libloader_app.so).
+  executable("loader_app") {
+    check_includes = false
+    output_dir = root_build_dir
+    sources = [
+      "//starboard/loader_app/loader_app.cc",
+      "//starboard/loader_app/loader_app_switches.cc",
+      "//starboard/loader_app/loader_app_switches.h",
+      "//starboard/loader_app/system_get_extension_shim.cc",
+      "//starboard/loader_app/system_get_extension_shim.h",
+    ]
+    deps = [
+      "//starboard/loader_app:common_loader_app_dependencies",
+      "//starboard:starboard_with_main",
+      "//starboard/content/fonts:copy_font_data",
+      "//starboard/elf_loader",
+    ]
+    data_deps = [ "//starboard/content/fonts:copy_font_data" ]
+  }
+}
diff --git a/starboard/elf_loader/BUILD.gn b/starboard/elf_loader/BUILD.gn
index e94ce0c2ed49..95b63f64ecd2 100644
--- a/starboard/elf_loader/BUILD.gn
+++ b/starboard/elf_loader/BUILD.gn
@@ -91,7 +91,8 @@ if (current_toolchain == starboard_toolchain) {
 
 # TODO: b/309493306 - Stop building evergreen targets for all non-evergreen platforms.
 if (current_toolchain == starboard_toolchain) {
-  target(starboard_level_final_executable_type, "elf_loader_sandbox") {
+  # Not a gtest target type, but it's used for loading test binaries mostly.
+  target(starboard_level_gtest_target_type, "elf_loader_sandbox") {
     output_dir = root_build_dir
 
     sources = [ "elf_loader_sandbox.cc" ]
@@ -116,7 +117,8 @@ if (current_toolchain == starboard_toolchain) {
 }
 
 if (current_toolchain == starboard_toolchain) {
-  target(starboard_level_final_executable_type, "elf_loader_sys_sandbox") {
+  # Not a gtest target type, but it's used for loading test binaries mostly.
+  target(starboard_level_gtest_target_type, "elf_loader_sys_sandbox") {
     # To properly function the system loader requires the starboard
     # symbols to be exported from the binary.
     # To allow symbols to be exported remove the '-fvisibility=hidden' flag
