--- base/debug/stack_trace.cc
+++ base/debug/stack_trace.cc
@@ -296,8 +296,10 @@
 }
 
 void StackTrace::OutputToStream(std::ostream* os) const {
-<<<<<<< HEAD
+// TODO: (cobalt b/398296821) Use stack_trace_starboard.cc or port stack_trace_posix.cc.
+#if !BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
   OutputToStreamWithPrefix(os, {});
+#endif
 }
 
 void StackTrace::OutputToStreamWithPrefix(std::ostream* os,
@@ -309,12 +311,6 @@
     return;
   }
   OutputToStreamWithPrefixImpl(os, prefix_string);
-=======
-// TODO: (cobalt b/398296821) Use stack_trace_starboard.cc or port stack_trace_posix.cc.
-#if !BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
-  OutputToStreamWithPrefix(os, nullptr);
-#endif
->>>>>>> c6f1d3da462 (Build base_unittests hermetically. (#4935))
 }
 
 std::string StackTrace::ToString() const {
