--- content/browser/BUILD.gn
+++ content/browser/BUILD.gn
@@ -2459,7 +2459,6 @@
   if (is_linux || is_chromeos) {
     # TODO: b/410019699 - Cobalt: Fix target dependencies on deps in this section
     if (!enable_cobalt_hermetic_hacks) {
-<<<<<<< HEAD
     sources += [
       "child_process_launcher_helper_linux.cc",
       "child_thread_type_switcher_linux.cc",
@@ -2481,27 +2480,6 @@
       "zygote_host/zygote_host_impl_linux.cc",
       "zygote_host/zygote_host_impl_linux.h",
     ]
-=======
-      sources += [
-        "child_process_launcher_helper_linux.cc",
-        "child_thread_type_switcher_linux.cc",
-        "child_thread_type_switcher_linux.h",
-        "font_access/font_enumeration_data_source_linux.cc",
-        "font_access/font_enumeration_data_source_linux.h",
-        "font_service.cc",
-        "font_service.h",
-        "media/video_encode_accelerator_provider_launcher.cc",
-        "media/video_encode_accelerator_provider_launcher.h",
-        "memory/swap_metrics_driver_impl_linux.cc",
-        "memory/swap_metrics_driver_impl_linux.h",
-        "sandbox_host_linux.cc",
-        "sandbox_host_linux.h",
-        "sandbox_ipc_linux.cc",
-        "sandbox_ipc_linux.h",
-        "zygote_host/zygote_host_impl_linux.cc",
-        "zygote_host/zygote_host_impl_linux.h",
-      ]
->>>>>>> 9984583f057 (content/ changes to build cobalt (#5517))
     }
 
     public_deps += [ "//components/services/font/public/mojom" ]
