--- content/utility/utility_main.cc
+++ content/utility/utility_main.cc
@@ -314,7 +314,6 @@
 #else
       NOTREACHED();
 #endif
-<<<<<<< HEAD
 #if BUILDFLAG(IS_LINUX)
     case sandbox::mojom::Sandbox::kVideoEffects:
 #if BUILDFLAG(ENABLE_VIDEO_EFFECTS)
@@ -324,23 +323,16 @@
       break;
 #endif  // BUILDFLAG(IS_LINUX)
 #if BUILDFLAG(USE_LINUX_VIDEO_ACCELERATION)
-=======
-#endif // !BUILDFLAG(IS_STARBOARD)
-#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_STARBOARD) || BUILDFLAG(IS_CHROMEOS_ASH)
->>>>>>> b4969c6295f ([POSIX] Remove dlopen and friends from Starboard (#6488))
     case sandbox::mojom::Sandbox::kHardwareVideoDecoding:
       pre_sandbox_hook =
           base::BindOnce(&media::HardwareVideoDecodingPreSandboxHook);
       break;
-<<<<<<< HEAD
-=======
-#endif  // BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_STARBOARD) || BUILDFLAG(IS_CHROMEOS_ASH)
->>>>>>> b4969c6295f ([POSIX] Remove dlopen and friends from Starboard (#6488))
     case sandbox::mojom::Sandbox::kHardwareVideoEncoding:
       pre_sandbox_hook =
           base::BindOnce(&media::HardwareVideoEncodingPreSandboxHook);
       break;
 #endif
+#endif  // !BUILDFLAG(IS_STARBOARD)
 #if BUILDFLAG(IS_CHROMEOS)
     case sandbox::mojom::Sandbox::kIme:
       pre_sandbox_hook = base::BindOnce(&ash::ime::ImePreSandboxHook);
