--- media/base/stream_parser_buffer.cc
+++ media/base/stream_parser_buffer.cc
@@ -65,15 +65,11 @@
                                        bool is_key_frame,
                                        Type type,
                                        TrackId track_id)
-<<<<<<< HEAD
-    : DecoderBuffer(data, data_size),
-=======
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
-    : DecoderBuffer(type, data, data_size, side_data, side_data_size),
+    : DecoderBuffer(type, data, data_size),
 #else // BUILDFLAG(USE_STARBOARD_MEDIA)
-    : DecoderBuffer(data, data_size, side_data, side_data_size),
+    : DecoderBuffer(data, data_size),
 #endif // BUILDFLAG(USE_STARBOARD_MEDIA)
->>>>>>> a86c930dc17 ([media] Improve DecoderBufferAllocator logging (#5036))
       decode_timestamp_(kNoDecodeTimestamp),
       config_id_(kInvalidConfigId),
       type_(type),
