--- cobalt/shell/browser/shell_content_browser_client.cc
+++ cobalt/shell/browser/shell_content_browser_client.cc
@@ -44,12 +44,9 @@
 #include "cobalt/shell/browser/shell_browser_main_parts.h"
 #include "cobalt/shell/browser/shell_devtools_manager_delegate.h"
 #include "cobalt/shell/browser/shell_web_contents_view_delegate_creator.h"
-<<<<<<< HEAD
-#include "cobalt/shell/common/shell_paths.h"
-=======
 #include "cobalt/shell/common/shell_controller.test-mojom.h"
+#include "cobalt/shell/common/shell_paths.h"
 #include "cobalt/shell/common/shell_switches.h"
->>>>>>> c15049941fb (Decouple cobalt/shell/common and cobalt/shell/resources from content shell (#6377))
 #include "components/custom_handlers/protocol_handler_registry.h"
 #include "components/custom_handlers/protocol_handler_throttle.h"
 #include "components/custom_handlers/simple_protocol_handler_registry_factory.h"
@@ -89,12 +86,6 @@
 #include "content/public/common/content_switches.h"
 #include "content/public/common/isolated_world_ids.h"
 #include "content/public/common/url_constants.h"
-<<<<<<< HEAD
-#include "content/shell/common/shell_controller.test-mojom.h"
-#include "content/shell/common/shell_switches.h"
-=======
-#include "content/public/common/user_agent.h"
->>>>>>> c15049941fb (Decouple cobalt/shell/common and cobalt/shell/resources from content shell (#6377))
 #include "media/mojo/buildflags.h"
 #include "media/mojo/mojom/media_service.mojom.h"
 #include "mojo/public/cpp/bindings/remote.h"
