--- skia/BUILD.gn
+++ skia/BUILD.gn
@@ -456,22 +456,11 @@
     sources += skia_ports_fci_sources
   }
 
-<<<<<<< HEAD
-  if (is_linux || is_chromeos || is_android) {
+  if (is_linux && !is_starboard || is_chromeos || is_android) {
     # Retain the files for the SkFontMgr_Android on linux to emulate android
     # fonts. See content/zygote/zygote_main_linux.cc
     # Note that this requires expat.
     sources += skia_ports_fontmgr_android_sources
-=======
-  if (is_linux && !is_starboard || is_chromeos || is_android) {
-    sources += [
-      # Retain the files for the SkFontMgr_Android on linux to emulate android
-      # fonts. See content/zygote/zygote_main_linux.cc
-      # Note that this requires expat.
-      "//third_party/skia/src/ports/SkFontMgr_android.cpp",
-      "//third_party/skia/src/ports/SkFontMgr_android_parser.cpp",
-    ]
->>>>>>> ecde1742f5a (Add a number of hermetic build targets (#5300))
   }
 
   if (is_fuchsia) {
