--- /usr/local/google/home/kaidokert/code/2ndproject/m120/63eb53c/conflict/BUILD.gn	2025-06-16 13:41:52.054987716 -0700
+++ /usr/local/google/home/kaidokert/code/2ndproject/m120/63eb53c/resolved/BUILD.gn	2025-06-16 13:42:31.983195998 -0700
@@ -831,18 +831,13 @@
       deps += [ "//components/cronet:cronet_tests" ]
     }
   }
-<<<<<<< HEAD
 
   if (is_chrome_for_testing) {
     deps += [ "//chrome/browser/chrome_for_testing:chrome_for_testing" ]
   }
 
-  if (chrobalt) {
-    deps += [ "//chrobalt:gn_all" ]
-=======
   if (is_cobalt) {
     deps += [ "//cobalt:gn_all" ]
->>>>>>> 63eb53c5c7561 (Rename Chrobalt to Cobalt. (#4276))
   }
 }
 
