diff -pu -r a/net/third_party/nss/ssl/sslreveal.c b/net/third_party/nss/ssl/sslreveal.c
--- a/net/third_party/nss/ssl/sslreveal.c	2012-04-25 07:50:12.000000000 -0700
+++ b/net/third_party/nss/ssl/sslreveal.c	2012-11-09 15:45:44.118267683 -0800
@@ -79,7 +79,6 @@ SSL_HandshakeNegotiatedExtension(PRFileD
   /* some decisions derived from SSL_GetChannelInfo */
   sslSocket * sslsocket = NULL;
   SECStatus rv = SECFailure;
-  PRBool enoughFirstHsDone = PR_FALSE;
 
   if (!pYes)
     return rv;
@@ -91,14 +90,8 @@ SSL_HandshakeNegotiatedExtension(PRFileD
     return rv;
   }
 
-  if (sslsocket->firstHsDone) {
-    enoughFirstHsDone = PR_TRUE;
-  } else if (sslsocket->ssl3.initialized && ssl3_CanFalseStart(sslsocket)) {
-    enoughFirstHsDone = PR_TRUE;
-  }
-
   /* according to public API SSL_GetChannelInfo, this doesn't need a lock */
-  if (sslsocket->opt.useSecurity && enoughFirstHsDone) {
+  if (sslsocket->opt.useSecurity) {
     if (sslsocket->ssl3.initialized) { /* SSL3 and TLS */
       /* now we know this socket went through ssl3_InitState() and
        * ss->xtnData got initialized, which is the only member accessed by
