commit f11613336a772057cd102a02759a6e4d111503d1
Author: Adam Langley <agl@chromium.org>
Date:   Mon Jun 20 15:52:10 2011 -0400

    renegoscsv.patch

diff --git a/mozilla/security/nss/lib/ssl/ssl3con.c b/mozilla/security/nss/lib/ssl/ssl3con.c
index e0cb4e9..455a532 100644
--- a/mozilla/security/nss/lib/ssl/ssl3con.c
+++ b/mozilla/security/nss/lib/ssl/ssl3con.c
@@ -3874,9 +3874,9 @@ ssl3_SendClientHello(sslSocket *ss)
     	return SECFailure;	/* ssl3_config_match_init has set error code. */
 
     /* HACK for SCSV in SSL 3.0.  On initial handshake, prepend SCSV,
-     * only if we're willing to complete an SSL 3.0 handshake.
+     * only if TLS is disabled.
      */
-    if (!ss->firstHsDone && ss->opt.enableSSL3) {
+    if (!ss->firstHsDone && !isTLS) {
 	/* Must set this before calling Hello Extension Senders, 
 	 * to suppress sending of empty RI extension.
 	 */
