--- third_party/fontconfig/include/config.h
+++ third_party/fontconfig/include/config.h
@@ -79,14 +79,6 @@
 
 #define HAVE_RANDOM 1
 
-<<<<<<< HEAD
-#if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
-/* TODO: (cobalt b/398295440) Add `random_r' support to Evergreen. */
-#define HAVE_RANDOM_R 0
-#else
-=======
-/* Define to 1 if you have the `random_r' function. */
->>>>>>> eb869eec366 (Cleanup fontconfig hack (#7363))
 #define HAVE_RANDOM_R 1
 
 #define HAVE_RAND_R 1
