--- build/android/gyp/compile_java.py
+++ build/android/gyp/compile_java.py
@@ -761,14 +761,11 @@
       # protobuf-generated files fail this check (javadoc has @deprecated,
       # but method missing @Deprecated annotation).
       '-Xlint:-dep-ann',
-<<<<<<< HEAD
       # Do not warn about finalize() methods. Android still intends to support
       # them.
       '-Xlint:-removal',
-=======
       # https://crbug.com/1441023
       '-J-XX:+PerfDisableSharedMem',
->>>>>>> f49b117fe3e (Cherry-pick a40b631 from chromium (#4881))
   ]
 
   if options.enable_errorprone:
