--- third_party/blink/renderer/core/dom/node.h
+++ third_party/blink/renderer/core/dom/node.h
@@ -36,10 +36,7 @@
 #include "third_party/blink/renderer/core/dom/node_rare_data.h"
 #include "third_party/blink/renderer/core/dom/tree_scope.h"
 #include "third_party/blink/renderer/core/style/computed_style_constants.h"
-<<<<<<< HEAD
 #include "third_party/blink/renderer/platform/bindings/exception_state.h"
-=======
->>>>>>> aff9c0340aa (BACKPORT: Pass HTMLMediaElement identifier to ::media::Renderer (#5463))
 #include "third_party/blink/renderer/platform/graphics/dom_node_id.h"
 #include "third_party/blink/renderer/platform/heap/collection_support/heap_hash_map.h"
 #include "third_party/blink/renderer/platform/heap/collection_support/heap_hash_set.h"
