--- base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_unittest.cc
+++ base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_unittest.cc
@@ -475,13 +475,8 @@
   ASSERT_EQ(0u, reinterpret_cast<uintptr_t>(pvalloc_ptr) % kPageSize);
   ASSERT_GE(allocs_intercepted_by_alignment[kPageSize], 1u);
   // pvalloc rounds the size up to the next page.
-<<<<<<< HEAD
   ASSERT_GE(allocs_intercepted_by_size[kPageSize], 1u);
-#endif  // PA_BUILDFLAG(IS_POSIX) && !PA_BUILDFLAG(IS_ANDROID)
-=======
-  ASSERT_GE(aligned_allocs_intercepted_by_size[kPageSize], 1u);
 #endif  // PA_BUILDFLAG(IS_POSIX) && !PA_BUILDFLAG(IS_ANDROID) && !PA_BUILDFLAG(IS_STARBOARD)
->>>>>>> 2627102ec98 (Build base_unittests hermetically. (#4935))
 
 #endif  // !PA_BUILDFLAG(IS_WIN) && !PA_BUILDFLAG(IS_APPLE)
 
