--- net/BUILD.gn
+++ net/BUILD.gn
@@ -1609,19 +1609,17 @@
     configs += [ "//build/config/compiler:optimize_max" ]
   }
 
-<<<<<<< HEAD
   if (!is_cronet_build) {
     deps += [ "//mojo/public/cpp/bindings:default_construct_tag" ]
   }
 
-=======
   if(is_starboard) {
     sources += [
       "cert/test_root_certs_builtin.cc",
       "cert/internal/system_trust_store_starboard.cc"
     ]
   }
->>>>>>> 50a799e08a7 (Disable NSS certificates (#5334))
+
   if (is_cobalt_hermetic_build) {
     sources += [
       "base/network_interfaces_starboard.cc",
