--- third_party/blink/renderer/modules/event_target_modules_names.json5
+++ third_party/blink/renderer/modules/event_target_modules_names.json5
@@ -93,12 +93,8 @@
     "VideoEncoder",
     "AudioDecoder",
     "VideoDecoder",
-<<<<<<< HEAD
     // TODO(cobalt, b/400518318): "Extract Cobalt event to separate file"
-=======
-    # TODO(cobalt, b/400518318): "Extract Cobalt event to separate file"
     "H5vccMetrics",
->>>>>>> 13da5ef47bd (blink: Add H5vccMetrics IDL, Blink implementation and Mojom (#5047))
     "H5vccRuntime",
   ],
 }
