--- build/config/ios/config.gni
+++ build/config/ios/config.gni
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-<<<<<<< HEAD
 import("//build/config/apple/mobile_config.gni")
 
 declare_args() {
@@ -12,44 +11,3 @@
   # application. It should generally NOT be enabled.
   ios_chrome_generate_order_file = false
 }
-=======
-import("//build/config/features.gni")  # For `use_blink`
-
-declare_args() {
-  # Configure the environment for which to build. Could be either "device",
-  # "simulator" or "catalyst". If unspecified, then it will be assumed to be
-  # "simulator" if the target_cpu is "x68" or "x64", "device" otherwise. The
-  # default is only there for compatibility reasons and will be removed (see
-  # crbug.com/1138425 for more details).
-  target_environment = ""
-
-  # Valid values: "iphoneos" (default), "tvos".
-  # Indicates the kind of iOS or iOS-based platform that is being targeted.
-  # Note that this value is available only when is_ios is also true (i.e. it
-  # cannot be used with the host toolchain).
-  target_platform = "iphoneos"
-}
-
-if (target_environment == "") {
-  if (current_cpu == "x86" || current_cpu == "x64") {
-    target_environment = "simulator"
-  } else {
-    target_environment = "device"
-  }
-}
-
-assert(target_environment == "simulator" || target_environment == "device" ||
-       target_environment == "catalyst")
-
-_target_platforms = [
-  "iphoneos",
-  "tvos",
-]
-assert(filter_include([ target_platform ], _target_platforms) != [],
-       "target_platform '$target_platform' does not match a target value: " +
-       "$_target_platforms")
-# Sanity check: tvOS builds require `use_blink`.
-if (target_platform == "tvos") {
-  assert(use_blink, "tvOS builds require use_blink=true")
-}
->>>>>>> e6e7e9c54f5 ([tvOS] Add minimum set of required backports + tvos-arm64-simulator build config (#7107))
