--- base/debug/debugger_posix.cc
+++ base/debug/debugger_posix.cc
@@ -57,10 +57,6 @@
 #include "base/strings/string_number_conversions.h"
 
 #if defined(USE_SYMBOLIZE)
-<<<<<<< HEAD
-=======
-// TODO(cobalt, b/377295011): remove the nogncheck annotation.
->>>>>>> 5c97a1227a94 (Fix/disable errors and enable gn check (#4518))
 #include "base/third_party/symbolize/symbolize.h"  // nogncheck
 #endif
 
