--- base/debug/dwarf_line_no.cc
+++ base/debug/dwarf_line_no.cc
@@ -22,15 +22,9 @@
 #include <unistd.h>
 
 #include "base/debug/buffered_dwarf_reader.h"
-<<<<<<< HEAD
-#include "base/third_party/symbolize/symbolize.h"
-#include "partition_alloc/pointers/raw_ptr.h"
-=======
-#include "base/debug/stack_trace.h"
-#include "base/memory/raw_ptr.h"
 // TODO(cobalt, b/377295011): remove the nogncheck annotation.
 #include "base/third_party/symbolize/symbolize.h"  // nogncheck
->>>>>>> 5c97a1227a94 (Fix/disable errors and enable gn check (#4518))
+#include "partition_alloc/pointers/raw_ptr.h"
 
 namespace base {
 namespace debug {
