--- cobalt/shell/BUILD.gn
+++ cobalt/shell/BUILD.gn
@@ -92,16 +92,6 @@
     "//third_party/blink/public:image_resources",
     "//third_party/blink/public:resources",
     "//third_party/blink/public/strings",
-<<<<<<< HEAD
-    "//ui/base",
-    "//ui/base/clipboard",
-    "//ui/base/ime/init",
-    "//ui/gfx",
-    "//ui/gfx/geometry",
-    "//ui/platform_window",
-=======
-    "//third_party/blink/public/strings:accessibility_strings",
->>>>>>> 5df296b991a (Remove redundant UI dependencies from ATV (#6364))
     "//url",
     "//v8",
   ]
