--- third_party/blink/renderer/platform/heap/thread_state_storage.h
+++ third_party/blink/renderer/platform/heap/thread_state_storage.h
@@ -43,15 +43,11 @@
 // Storage for all ThreadState objects. This includes the main-thread
 // ThreadState as well. Keep it outside the class so that PLATFORM_EXPORT
 // doesn't apply to it (otherwise, clang-cl complains).
-<<<<<<< HEAD
-extern constinit thread_local ThreadStateStorage* g_thread_specific_
-=======
 #if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
 // TODO: b/412388549 - Cobalt: Fix this hack.
 extern ThreadStateStorage* g_thread_specific_;
 #else  // BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
-extern thread_local ThreadStateStorage* g_thread_specific_ CONSTINIT
->>>>>>> bbe826d5f2e (Build libcobalt.so hermetically (#5048))
+extern constinit thread_local ThreadStateStorage* g_thread_specific_
     __attribute__((tls_model(BLINK_HEAP_THREAD_LOCAL_MODEL)));
 #endif  // BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
 
