--- base/BUILD.gn
+++ base/BUILD.gn
@@ -5442,7 +5442,6 @@
   }
 }
 
-<<<<<<< HEAD
 process_version("check_version_internal") {
   template_file = "//base/check_version_internal.h.in"
   output = "$target_gen_dir/check_version_internal.h"
@@ -5469,12 +5468,13 @@
       "-e",
       "NEXT_VERSION_MAJOR=\"9999\"",
     ]
-=======
+  }
+}
+
 if (is_cobalt_hermetic_build) {
   # TODO: b/413130400 - Cobalt: Remove this config 
   # after removing base/starboard/linker_stub.h
   config("base_starboard_linker_stub") {
     cflags = ["-Wno-unreachable-code-break"]
->>>>>>> dbf055e9533 (Fix stub linker error (#5558))
   }
 }
