--- content/browser/renderer_host/render_process_host_impl.cc
+++ content/browser/renderer_host/render_process_host_impl.cc
@@ -5552,13 +5552,9 @@
     // Not all platforms launch processes in the same backgrounded state. Make
     // sure |priority_.visible| reflects this platform's initial process
     // state.
-<<<<<<< HEAD
-#if BUILDFLAG(IS_APPLE) && !BUILDFLAG(IS_IOS_TVOS)
-=======
 #if BUILDFLAG(IS_STARBOARD)
     priority_.visible = true;
-#elif BUILDFLAG(IS_APPLE)
->>>>>>> 9346cfcd213 (Fix remaining compiler issues in hermetic cobalt (#5316))
+#elif BUILDFLAG(IS_APPLE) && !BUILDFLAG(IS_IOS_TVOS)
     priority_.visible = child_process_launcher_->GetProcess().GetPriority(
                             ChildProcessTaskPortProvider::GetInstance()) ==
                         base::Process::Priority::kUserBlocking;
