--- tools/gritsettings/resource_ids.spec
+++ tools/gritsettings/resource_ids.spec
@@ -1525,7 +1525,6 @@
     "includes": [10060],
   },
 
-<<<<<<< HEAD
   "<(SHARED_INTERMEDIATE_DIR)/third_party/blink/public/strings/permission_element_generated_strings.grd": {
     "META": {"sizes": {"messages": [2000],}},
     "messages": [10080],
@@ -1533,14 +1532,12 @@
 
   "third_party/search_engines_data/resources/search_engines_scaled_resources.grd": {
     "structures": [10100],
-  }
+  },
 
-=======
   "cobalt/shell/shell_resources.grd": {
     "includes": [31500]
   },
 
->>>>>>> c15049941fb (Decouple cobalt/shell/common and cobalt/shell/resources from content shell (#6377))
   # END "everything else" section.
   # Everything but chrome/, components/, content/, and ios/
 
