--- content/public/android/java/src/org/chromium/content/browser/webcontents/WebContentsImpl.java
+++ content/public/android/java/src/org/chromium/content/browser/webcontents/WebContentsImpl.java
@@ -606,16 +606,6 @@
         WebContentsImplJni.get().collapseSelection(mNativeWebContentsAndroid);
     }
 
-<<<<<<< HEAD
-=======
-    @Override
-    public void onHide() {
-        checkNotDestroyed();
-        SelectionPopupControllerImpl controller = getSelectionPopupController();
-        if (controller != null) controller.hidePopupsAndPreserveSelection();
-        WebContentsImplJni.get().onHide(mNativeWebContentsAndroid);
-    }
-
     @Override
     public void onFreeze() {
         checkNotDestroyed();
@@ -628,15 +618,6 @@
         WebContentsImplJni.get().onResume(mNativeWebContentsAndroid);
     }
 
-    @Override
-    public void onShow() {
-        checkNotDestroyed();
-        SelectionPopupControllerImpl controller = getSelectionPopupController();
-        if (controller != null) controller.restoreSelectionPopupsIfNecessary();
-        WebContentsImplJni.get().onShow(mNativeWebContentsAndroid);
-    }
-
->>>>>>> 206d5743ac9 (Add freeze and resume events on suspend/resume (#5613))
     private SelectionPopupControllerImpl getSelectionPopupController() {
         return SelectionPopupControllerImpl.fromWebContents(this);
     }
@@ -1342,17 +1323,10 @@
 
         void collapseSelection(long nativeWebContentsAndroid);
 
-<<<<<<< HEAD
-=======
-        void onHide(long nativeWebContentsAndroid);
-
-        void onShow(long nativeWebContentsAndroid);
-
         void onFreeze(long nativeWebContentsAndroid);
 
         void onResume(long nativeWebContentsAndroid);
 
->>>>>>> 206d5743ac9 (Add freeze and resume events on suspend/resume (#5613))
         void setImportance(long nativeWebContentsAndroid, int importance);
 
         void suspendAllMediaPlayers(long nativeWebContentsAndroid);
