--- base/BUILD.gn
+++ base/BUILD.gn
@@ -2220,106 +2220,6 @@
     }
   }
 
-<<<<<<< HEAD
-=======
-  # iOS
-  if (is_ios) {
-    sources += [
-      "base_paths_mac.h",
-      "base_paths_mac.mm",
-      "critical_closure_internal_ios.mm",
-      "ios/block_types.h",
-      "ios/device_util.h",
-      "ios/device_util.mm",
-      "ios/ios_util.h",
-      "ios/ios_util.mm",
-      "ios/ns_error_util.h",
-      "ios/ns_error_util.mm",
-      "ios/ns_range.h",
-      "ios/scoped_critical_action.h",
-      "ios/scoped_critical_action.mm",
-      "mac/bridging.h",
-      "native_library_ios.mm",
-      "power_monitor/power_monitor_device_source_ios.mm",
-      "process/process_metrics_ios.cc",
-      "profiler/stack_sampler_ios.cc",
-      "system/sys_info_ios.mm",
-    ]
-
-    if (use_blink) {
-      sources += [
-        "files/file_path_watcher_kqueue.cc",
-        "files/file_path_watcher_kqueue.h",
-        "files/file_path_watcher_mac.cc",
-        "ios/sim_header_shims.h",
-        "mac/mach_port_rendezvous.cc",
-        "mac/mach_port_rendezvous.h",
-        "process/kill_mac.cc",
-        "process/launch_mac.cc",
-        "process/memory_mac.mm",
-        "process/port_provider_mac.cc",
-        "process/port_provider_mac.h",
-        "process/process_handle_mac.cc",
-        "process/process_iterator_ios.mm",
-        "process/process_mac.cc",
-        "process/process_metrics_posix.cc",
-        "process/process_posix.cc",
-        "sync_socket_posix.cc",
-        "synchronization/waitable_event_watcher_mac.cc",
-      ]
-      libs += [ "bsm" ]
-    } else {
-      sources += [
-        "process/launch_ios.cc",
-        "process/memory_stubs.cc",
-        "process/process_stubs.cc",
-      ]
-    }
-
-    if (is_cronet_build) {
-      sources += [
-        "message_loop/message_pump_io_ios.cc",
-        "message_loop/message_pump_io_ios.h",
-      ]
-    } else {
-      sources += [
-        "message_loop/message_pump_kqueue.cc",
-        "message_loop/message_pump_kqueue.h",
-      ]
-    }
-
-    if (ios_stack_profiler_enabled) {
-      sources += [
-        "profiler/frame_pointer_unwinder.cc",
-        "profiler/frame_pointer_unwinder.h",
-        "profiler/suspendable_thread_delegate_mac.cc",
-        "profiler/suspendable_thread_delegate_mac.h",
-      ]
-    }
-    if (use_allocator_shim) {
-      sources += [
-        "allocator/partition_allocator/shim/allocator_interception_mac.h",
-        "allocator/partition_allocator/shim/allocator_interception_mac.mm",
-        "allocator/partition_allocator/shim/malloc_zone_functions_mac.cc",
-        "allocator/partition_allocator/shim/malloc_zone_functions_mac.h",
-      ]
-    } else {
-      sources -= [
-        "sampling_heap_profiler/poisson_allocation_sampler.cc",
-        "sampling_heap_profiler/poisson_allocation_sampler.h",
-        "sampling_heap_profiler/sampling_heap_profiler.cc",
-        "sampling_heap_profiler/sampling_heap_profiler.h",
-      ]
-    }
-
-    if (current_cpu == "x64" || current_cpu == "arm64") {
-      sources += [ "time/time_exploded_posix.cc" ]
-    } else {
-      sources += [ "time/time_exploded_ios.cc" ]
-    }
-
-    frameworks += [ "UIKit.framework" ]
-  }
 
   if (is_cobalt) {
     sources += [
@@ -2328,7 +2228,6 @@
       ]
   }
 
->>>>>>> 1c40ffd4dde9 (Chrobalt ATV UA String Implementation With No Dependencies on SB (#4388))
   if (use_blink) {
     sources += [
       "files/file_path_watcher.cc",
