--- content/renderer/media/media_factory.h
+++ content/renderer/media/media_factory.h
@@ -133,11 +133,7 @@
       blink::WebURL url,
       const RenderFrameMediaPlaybackOptions& renderer_media_playback_options,
       media::DecoderFactory* decoder_factory,
-<<<<<<< HEAD
       media::RemotePlaybackClientWrapper* client_wrapper,
-=======
-      std::unique_ptr<media::RemotePlaybackClientWrapper> client_wrapper,
->>>>>>> 80e27f0e4c0 (BACKPORT: Pass HTMLMediaElement identifier to ::media::Renderer (#5463))
       base::WeakPtr<media::MediaObserver>* out_media_observer,
       int element_id);
 
