--- v8/src/base/cpu.cc
+++ v8/src/base/cpu.cc
@@ -22,11 +22,7 @@
 #if V8_OS_QNX
 #include <sys/syspage.h>  // cpuinfo
 #endif
-<<<<<<< HEAD
-#if V8_OS_LINUX && V8_HOST_ARCH_PPC64
-=======
-#if V8_OS_LINUX && (V8_HOST_ARCH_PPC || V8_HOST_ARCH_PPC64 || BUILDFLAG(IS_STARBOARD))
->>>>>>> 445476a7299 (Build raspi-2 hermetically. (#6216))
+#if V8_OS_LINUX && (V8_HOST_ARCH_PPC64 || BUILDFLAG(IS_STARBOARD))
 #include <elf.h>
 #endif
 #if V8_OS_AIX
