--- components/viz/service/display/overlay_processor_ozone.cc
+++ components/viz/service/display/overlay_processor_ozone.cc
@@ -233,14 +233,10 @@
     std::unique_ptr<PixmapProvider> pixmap_provider)
     : overlay_candidates_(std::move(overlay_candidates)),
       available_strategies_(std::move(available_strategies)),
-<<<<<<< HEAD
       pixmap_provider_(std::move(pixmap_provider)) {
-=======
-      shared_image_interface_(shared_image_interface) {
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
   strategies_.push_back(std::make_unique<OverlayStrategyUnderlayStarboard>(this));
 #else // BUILDFLAG(USE_STARBOARD_MEDIA)
->>>>>>> 3945fc3c659 ([linux] Support SbPlayer via punching a video hole frame (#4889))
   for (OverlayStrategy strategy : available_strategies_) {
     switch (strategy) {
       case OverlayStrategy::kFullscreen:
