--- content/public/common/content_features.cc
+++ content/public/common/content_features.cc
@@ -1410,15 +1410,11 @@
 BASE_FEATURE(kWebViewSuppressTapDuringFling,
              "WebViewSuppressTapDuringFling",
              base::FEATURE_ENABLED_BY_DEFAULT);
-<<<<<<< HEAD
-=======
-
 BASE_FEATURE(kDoNotGenerateChromiumA11yTree,
              "DoNotGenerateChromiumA11yTree",
              BUILDFLAG(IS_COBALT) ? base::FEATURE_ENABLED_BY_DEFAULT
                                    : base::FEATURE_DISABLED_BY_DEFAULT);
 
->>>>>>> 85a0545f88b (Stop creating a11y node provder for Chrobalt (#5271))
 #endif  // BUILDFLAG(IS_ANDROID)
 
 #if BUILDFLAG(IS_MAC)
