--- third_party/blink/renderer/platform/heap/thread_state_storage.h
+++ third_party/blink/renderer/platform/heap/thread_state_storage.h
@@ -43,15 +43,7 @@
 // Storage for all ThreadState objects. This includes the main-thread
 // ThreadState as well. Keep it outside the class so that PLATFORM_EXPORT
 // doesn't apply to it (otherwise, clang-cl complains).
-<<<<<<< HEAD
-#if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
-// TODO: b/412388549 - Cobalt: Fix this hack.
-extern ThreadStateStorage* g_thread_specific_;
-#else  // BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
 extern constinit thread_local ThreadStateStorage* g_thread_specific_
-=======
-extern thread_local ThreadStateStorage* g_thread_specific_ CONSTINIT
->>>>>>> dfb264009ef (Fix the Blink heap tls model for Evergreen. (#5583))
     __attribute__((tls_model(BLINK_HEAP_THREAD_LOCAL_MODEL)));
 
 // ThreadStateStorage is the explicitly managed TLS- and global-backed storage
