--- third_party/webrtc/rtc_base/platform_thread_types.cc
+++ third_party/webrtc/rtc_base/platform_thread_types.cc
@@ -34,16 +34,12 @@
 #endif
 // IWYU pragma: end_keep
 
-<<<<<<< HEAD
-namespace webrtc {
-=======
 #include "build/build_config.h"
 #if BUILDFLAG(IS_STARBOARD)
 #include "starboard/thread.h"  // nogncheck
 #endif
 
-namespace rtc {
->>>>>>> 9346cfcd213 (Fix remaining compiler issues in hermetic cobalt (#5316))
+namespace webrtc {
 
 PlatformThreadId CurrentThreadId() {
 #if defined(WEBRTC_WIN)
