--- /usr/local/google/home/kaidokert/code/2ndproject/m120/ac080af/conflict/BUILD.gn	2025-06-16 13:12:56.112544646 -0700
+++ /usr/local/google/home/kaidokert/code/2ndproject/m120/ac080af/resolved/BUILD.gn	2025-06-16 13:13:38.220850737 -0700
@@ -831,14 +831,13 @@
       deps += [ "//components/cronet:cronet_tests" ]
     }
   }
-<<<<<<< HEAD
 
   if (is_chrome_for_testing) {
     deps += [ "//chrome/browser/chrome_for_testing:chrome_for_testing" ]
-=======
+  }
+
   if (chrobalt) {
     deps += [ "//chrobalt:gn_all" ]
->>>>>>> ac080afddd6db (Build starboard shared library for linux-x64x11 (#4215))
   }
 }
 
