diff --git a/openssl/crypto/bio/bss_dgram.c b/openssl/crypto/bio/bss_dgram.c
index 1b1e4be..3c66dd1 100644
--- a/openssl/crypto/bio/bss_dgram.c
+++ b/openssl/crypto/bio/bss_dgram.c
@@ -447,7 +447,7 @@ static long dgram_ctrl(BIO *b, int cmd, long num, void *ptr)
 	bio_dgram_data *data = NULL;
 #if defined(IP_MTU_DISCOVER) || defined(IP_MTU)
 	long sockopt_val = 0;
-	unsigned int sockopt_len = 0;
+	socklen_t sockopt_len = 0;
 #endif
 #ifdef OPENSSL_SYS_LINUX
 	socklen_t addr_len;
diff --git a/openssl/crypto/cryptlib.c b/openssl/crypto/cryptlib.c
index 766ea8c..50fd492 100644
--- a/openssl/crypto/cryptlib.c
+++ b/openssl/crypto/cryptlib.c
@@ -504,7 +504,7 @@ void CRYPTO_THREADID_current(CRYPTO_THREADID *id)
 	CRYPTO_THREADID_set_numeric(id, (unsigned long)find_thread(NULL));
 #else
 	/* For everything else, default to using the address of 'errno' */
-	CRYPTO_THREADID_set_pointer(id, &errno);
+	CRYPTO_THREADID_set_pointer(id, (void*)&errno);
 #endif
 	}
 
diff --git a/openssl/ssl/ssl_lib.c b/openssl/ssl/ssl_lib.c
index 8340854..d6b8ca8 100644
--- a/openssl/ssl/ssl_lib.c
+++ b/openssl/ssl/ssl_lib.c
@@ -160,6 +160,9 @@
 #include <openssl/engine.h>
 #endif
 
+/* defined in ssl_ciph.c - not exported by any header. */
+extern const char* SSL_CIPHER_authentication_method(const SSL_CIPHER* cipher);
+
 const char *SSL_version_str=OPENSSL_VERSION_TEXT;
 
 SSL3_ENC_METHOD ssl3_undef_enc_method={
