--- build/android/java/templates/BuildConfig.template
+++ build/android/java/templates/BuildConfig.template
@@ -87,16 +87,15 @@
     public static MAYBE_FINAL boolean IS_FOR_TEST MAYBE_FALSE;
 #endif
 
-<<<<<<< HEAD
 #if defined(_WRITE_CLANG_PROFILING_DATA)
     public static MAYBE_FINAL boolean WRITE_CLANG_PROFILING_DATA = true;
 #else
     public static MAYBE_FINAL boolean WRITE_CLANG_PROFILING_DATA MAYBE_FALSE;
-=======
+#endif
+
 #if defined(_DISABLE_DEBUG_LOGS)
     public static MAYBE_FINAL boolean ENABLE_DEBUG_LOGS MAYBE_FALSE;
 #else
     public static MAYBE_FINAL boolean ENABLE_DEBUG_LOGS = true;
->>>>>>> 79526f42c24 (Android: Fix VERBOSE and DEBUG logs not appearing in debug builds)
 #endif
 }
