--- third_party/blink/renderer/modules/BUILD.gn
+++ third_party/blink/renderer/modules/BUILD.gn
@@ -683,17 +683,10 @@
     ]
   }
 
-<<<<<<< HEAD
-=======
   if (is_cobalt) {
     sources += [ "cobalt/h5vcc_experiments/experiments_utils_test.cc" ]
   }
 
-  if (media_use_openh264) {
-    sources += [ "mediarecorder/h264_encoder_unittest.cc" ]
-  }
-
->>>>>>> 0fe5c8ee97d (Add tests for h_5_vcc_experiments. (#5773))
   configs += [
     "//third_party/blink/renderer:config",
     "//third_party/blink/renderer:inside_blink",
