--- DEPS
+++ DEPS
@@ -519,11 +519,7 @@
   'libcxx_revision':       'a429c26ae25c26a569ff12390d5f9be70c5e286b',
 
   # GN CIPD package version.
-<<<<<<< HEAD
   'gn_version': 'git_revision:e4702d7409069c4f12d45ea7b7f0890717ca3f4b',
-=======
-  'gn_version': 'git_revision:8ea3af9118a65387fe4d8428db6f59c5927dd55a',  # Cobalt: keep until updating past mid-2025
->>>>>>> b19b7315539 (Clean up runtime deps for modular build  (#6066))
 
   # ninja CIPD package version.
   # https://chrome-infra-packages.appspot.com/p/infra/3pp/tools/ninja
