--- third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h
+++ third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h
@@ -266,7 +266,6 @@
 //                                        deprecated; calling a marked function
 //                                        should generate a compiler warning
 
-<<<<<<< HEAD
 // The definition of GTEST_INTERNAL_CPLUSPLUS_LANG comes first because it can
 // potentially be used as an #include guard.
 #if defined(_MSVC_LANG)
@@ -295,9 +294,8 @@
 #elif (!defined(__has_include) || GTEST_INTERNAL_HAS_INCLUDE(<ciso646>))
 #include <ciso646>  // Pre-C++20
 #endif
-=======
+
 #include "build/build_config.h"
->>>>>>> 58e37190fa8 (Starboardize googletest to run nplb in hermetic builds (#4962))
 
 #include <ctype.h>   // for isspace, etc
 #include <stddef.h>  // for ptrdiff_t
@@ -1255,10 +1253,6 @@
 
 #endif  // GTEST_HAS_DEATH_TEST
 
-<<<<<<< HEAD
-// Defines synchronization primitives.
-#ifdef GTEST_IS_THREADSAFE
-=======
 // TODO: b/399507045 - Cobalt: Fix build error, remove hack
 #if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
 // Define synchronization primitives.
@@ -1358,8 +1352,9 @@
 };
 
 #else  // BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
-#if GTEST_IS_THREADSAFE
->>>>>>> 58e37190fa8 (Starboardize googletest to run nplb in hermetic builds (#4962))
+
+// Defines synchronization primitives.
+#ifdef GTEST_IS_THREADSAFE
 
 #ifdef GTEST_OS_WINDOWS
 // Provides leak-safe Windows kernel handle ownership.
@@ -2156,13 +2151,6 @@
 
 namespace posix {
 
-<<<<<<< HEAD
-// File system porting.
-// Note: Not every I/O-related function is related to file systems, so don't
-// just disable all of them here. For example, fileno() and isatty(), etc. must
-// always be available in order to detect if a pipe points to a terminal.
-#ifdef GTEST_OS_WINDOWS
-=======
 // TODO: b/399507045 - Cobalt: Fix build error, remove hack
 #if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
 
@@ -2250,10 +2238,12 @@
 
 #else // BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
 
-// Functions with a different name on Windows.
 
-#if GTEST_OS_WINDOWS
->>>>>>> 58e37190fa8 (Starboardize googletest to run nplb in hermetic builds (#4962))
+// File system porting.
+// Note: Not every I/O-related function is related to file systems, so don't
+// just disable all of them here. For example, fileno() and isatty(), etc. must
+// always be available in order to detect if a pipe points to a terminal.
+#ifdef GTEST_OS_WINDOWS
 
 typedef struct _stat StatStruct;
 
@@ -2499,19 +2489,8 @@
 #define GTEST_FLAG(name) FLAGS_gtest_##name
 #endif  // !defined(GTEST_FLAG)
 
-<<<<<<< HEAD
 // Pick a command line flags implementation.
 #ifdef GTEST_INTERNAL_HAS_ABSL_FLAGS
-=======
-#if !defined(GTEST_USE_OWN_FLAGFILE_FLAG_)
-// TODO: b/399507045 - Cobalt: Investigate and remove if unnecessary
-#if BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
-#define GTEST_USE_OWN_FLAGFILE_FLAG_ 0
-#else
-#define GTEST_USE_OWN_FLAGFILE_FLAG_ 1
-#endif // BUILDFLAG(ENABLE_COBALT_HERMETIC_HACKS)
-#endif  // !defined(GTEST_USE_OWN_FLAGFILE_FLAG_)
->>>>>>> 58e37190fa8 (Starboardize googletest to run nplb in hermetic builds (#4962))
 
 // Macros for defining flags.
 #define GTEST_DEFINE_bool_(name, default_val, doc) \
