--- content/shell/renderer/shell_content_renderer_client.cc
+++ content/shell/renderer/shell_content_renderer_client.cc
@@ -54,20 +54,18 @@
 #include "media/base/media_switches.h"
 #endif
 
-<<<<<<< HEAD
 #if (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)) && \
     (defined(ARCH_CPU_X86_64) || defined(ARCH_CPU_ARM64))
 #define ENABLE_WEB_ASSEMBLY_TRAP_HANDLER_LINUX
 #include "base/debug/stack_trace.h"
 #include "v8/include/v8-wasm-trap-handler-posix.h"
 #endif
-=======
+
 #if BUILDFLAG(IS_COBALT)
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
 #include "starboard/media.h"
 #endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
 #endif  // BUILDFLAG(IS_COBALT)
->>>>>>> f33a6e4c1ddc ([media] Integrate SbMediaCanPlayMimeAndKeySystem() (#4340))
 
 namespace content {
 
