--- ui/gfx/BUILD.gn
+++ ui/gfx/BUILD.gn
@@ -646,15 +646,11 @@
     "//ui/gfx/geometry",
   ]
 
-<<<<<<< HEAD
   if (!is_nacl) {
     deps += [ ":native_widget_types_sources" ]
   }
 
-  if (is_linux || is_chromeos) {
-=======
   if (is_linux && !is_starboard || is_chromeos) {
->>>>>>> 79b9f4332a5 (Add a number of hermetic build targets (#5300))
     sources += [
       "linux/client_native_pixmap_dmabuf.cc",
       "linux/client_native_pixmap_dmabuf.h",
