--- ui/gl/gl_features.cc
+++ ui/gl/gl_features.cc
@@ -28,17 +28,7 @@
 namespace features {
 namespace {
 
-<<<<<<< HEAD
-#if BUILDFLAG(IS_ANDROID)
-=======
-#if BUILDFLAG(IS_APPLE)
-BASE_FEATURE(kGpuVsync, "GpuVsync", base::FEATURE_DISABLED_BY_DEFAULT);
-#else
-BASE_FEATURE(kGpuVsync, "GpuVsync", base::FEATURE_ENABLED_BY_DEFAULT);
-#endif
-
 #if BUILDFLAG(IS_ANDROID) && BUILDFLAG(ENABLE_VALIDATING_COMMAND_DECODER)
->>>>>>> d6d0e95447a (Add the missing ENABLE_VALIDATING_COMMAND_DECODER build flag (#5439))
 const base::FeatureParam<std::string>
     kPassthroughCommandDecoderBlockListByBrand{
         &kDefaultPassthroughCommandDecoder, "BlockListByBrand", ""};
