--- third_party/blink/renderer/modules/BUILD.gn
+++ third_party/blink/renderer/modules/BUILD.gn
@@ -332,19 +332,7 @@
     rebase_path(outputs[0], root_build_dir),
   ]
 
-<<<<<<< HEAD
   deps = [ "//third_party/blink/renderer/bindings:web_idl_database" ]
-=======
-  if (is_cobalt) {
-    sources += [
-      "cobalt/h5vcc_runtime/deep_link_event.idl",
-      "cobalt/h5vcc_system/h_5_vcc_system.idl",
-    ]
-  }
-
-  output_file = "modules/event_interface_modules_names.json5"
-  suffix = "Modules"
->>>>>>> 111eefe8c0e (Add h5vcc.system.concealOrStop(). (#4926))
 }
 
 # make_event_factory -----------------------------------------------------------
