--- build/config/android/rules.gni
+++ build/config/android/rules.gni
@@ -1944,14 +1944,13 @@
         defines += [ "_IS_FOR_TEST" ]
       }
 
-<<<<<<< HEAD
       if (defined(invoker.write_clang_profiling_data) &&
           invoker.write_clang_profiling_data) {
         defines += [ "_WRITE_CLANG_PROFILING_DATA" ]
-=======
+      }
+
       if (!is_java_debug && !_test_only) {
         defines += [ "_DISABLE_DEBUG_LOGS" ]
->>>>>>> 79526f42c24 (Android: Fix VERBOSE and DEBUG logs not appearing in debug builds)
       }
     }
   }
