--- third_party/blink/renderer/core/html/media/html_media_element_test.cc
+++ third_party/blink/renderer/core/html/media/html_media_element_test.cc
@@ -1686,7 +1686,6 @@
   EXPECT_FALSE(MediaIsPlaying());
 }
 
-<<<<<<< HEAD
 TEST_P(HTMLMediaElementTest, MoveToAnotherDocument) {
   auto* second_document =
       dummy_page_holder_->GetDocument().implementation().createHTMLDocument();
@@ -2201,7 +2200,7 @@
       video->GetDocument(), event_type_names::kFullscreenchange,
       video->visibility_tracker_for_tests()));
 }
-=======
+
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
 TEST(HTMLMediaElementTest, CanHandleCobaltProgressiveSupportQueries) {
   const ContentType progressive_type(
@@ -2239,6 +2238,5 @@
   }
 }
 #endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
->>>>>>> d9ec2ba2006a (Disable progressive playback support by default (#4547))
 
 }  // namespace blink
