--- DEPS
+++ DEPS
@@ -519,11 +519,7 @@
   'libcxx_revision':       'a429c26ae25c26a569ff12390d5f9be70c5e286b',
 
   # GN CIPD package version.
-<<<<<<< HEAD
   'gn_version': 'git_revision:e4702d7409069c4f12d45ea7b7f0890717ca3f4b',
-=======
-  'gn_version': 'git_revision:5a004f9427a050c6c393c07ddb85cba8ff3849fa',
->>>>>>> b25e8607e6a (Revert "Clean up runtime deps for modular build " (#6167))
 
   # ninja CIPD package version.
   # https://chrome-infra-packages.appspot.com/p/infra/3pp/tools/ninja
