--- content/utility/utility_blink_platform_with_sandbox_support_impl.h
+++ content/utility/utility_blink_platform_with_sandbox_support_impl.h
@@ -33,12 +33,8 @@
   blink::WebSandboxSupport* GetSandboxSupport() override;
 
  private:
-<<<<<<< HEAD
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_MAC) || \
+#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_MAC) || \
     BUILDFLAG(IS_WIN)
-=======
-#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_MAC)
->>>>>>> 632b3464a30 (Feature/fonts (#7049))
   std::unique_ptr<blink::WebSandboxSupport> sandbox_support_;
 #endif
 };
