--- base/allocator/partition_allocator/src/partition_alloc/partition_alloc_unittest.cc
+++ base/allocator/partition_allocator/src/partition_alloc/partition_alloc_unittest.cc
@@ -57,22 +57,13 @@
 #include <arm_acle.h>
 #endif
 
-<<<<<<< HEAD
 #if PA_BUILDFLAG(IS_POSIX)
-#if PA_BUILDFLAG(IS_LINUX)
-=======
-#if BUILDFLAG(IS_POSIX)
-#if BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_STARBOARD)
->>>>>>> 61f3855da9c (Build base_unittests hermetically. (#4935))
+#if PA_BUILDFLAG(IS_LINUX) && !PA_BUILDFLAG(IS_STARBOARD)
 // We need PKEY_DISABLE_WRITE in this file; glibc defines it in sys/mman.h but
 // it's actually Linux-specific and other Linux libcs define it in linux/mman.h.
 // We have to include both to be sure we get the definition.
 #include <linux/mman.h>
-<<<<<<< HEAD
-#endif  // PA_BUILDFLAG(IS_LINUX)
-=======
-#endif  // BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_STARBOARD)
->>>>>>> 61f3855da9c (Build base_unittests hermetically. (#4935))
+#endif  // PA_BUILDFLAG(IS_LINUX) && !PA_BUILDFLAG(IS_STARBOARD)
 #include <sys/mman.h>
 #include <sys/resource.h>
 #include <sys/time.h>
