--- content/renderer/renderer_blink_platform_impl.cc
+++ content/renderer/renderer_blink_platform_impl.cc
@@ -524,20 +524,12 @@
 
 bool RendererBlinkPlatformImpl::DecodeAudioFileData(
     blink::WebAudioBus* destination_bus,
-<<<<<<< HEAD
     base::span<const char> audio_file_data) {
-  return content::DecodeAudioFileData(destination_bus, audio_file_data);
-=======
-    const char* audio_file_data,
-    size_t data_size) {
 #if BUILDFLAG(IS_COBALT)
-  return cobalt::DecodeAudioFileData(destination_bus, audio_file_data,
-                                      data_size);
+  return cobalt::DecodeAudioFileData(destination_bus, audio_file_data);
 #else // BUILDFLAG(IS_COBALT)
-  return content::DecodeAudioFileData(destination_bus, audio_file_data,
-                                      data_size);
+  return content::DecodeAudioFileData(destination_bus, audio_file_data);
 #endif // BUILDFLAG(IS_COBALT)
->>>>>>> fb577e3f745 (Add WavAudioHandler to WebAudio (#5049))
 }
 
 //------------------------------------------------------------------------------
