--- third_party/blink/renderer/platform/audio/audio_destination.cc
+++ third_party/blink/renderer/platform/audio/audio_destination.cc
@@ -63,7 +63,6 @@
 
 namespace {
 
-<<<<<<< HEAD
 // This FIFO size of 16,384 was chosend based on the UMA data. It's the nearest
 // multiple of 128 to 16,354 sample-frames, which represents 100% of the
 // histogram from "WebAudio.AudioDestination.HardwareBufferSize".
@@ -71,21 +70,6 @@
 // audio device report a large buffer size. This redundancy allows such device
 // to play audio via Web Audio API.
 constexpr uint32_t kFIFOSize = 128 * 128;
-=======
-// FIFO Size.
-//
-// TODO(hongchan): This was estimated based on the largest callback buffer size
-// that we would ever need. The current UMA stats indicates that this is, in
-// fact, probably too small. There are Android devices out there with a size of
-// 8000 or so.  We might need to make this larger. See: crbug.com/670747
-#if BUILDFLAG(IS_COBALT)
-// TODO(b/376903291): The change will not be needed after we rebase to m126 or
-// later version.
-constexpr uint32_t kFIFOSize = 128 * 128;
-#else   // BUILDFLAG(IS_COBALT)
-constexpr uint32_t kFIFOSize = 96 * 128;
-#endif  // BUILDFLAG(IS_COBALT)
->>>>>>> c711a03c967b ([android] Increase kFIFOSize in AudioDestination (#4376))
 
 const char* DeviceStateToString(AudioDestination::DeviceState state) {
   switch (state) {
