--- DEPS
+++ DEPS
@@ -2896,22 +2896,7 @@
       'dep_type': 'cipd',
   },
 
-<<<<<<< HEAD
-  'src/third_party/android_deps/cipd/libs/com_google_android_gms_play_services_ads_identifier': {
-      'packages': [
-          {
-              'package': 'chromium/third_party/android_deps/libs/com_google_android_gms_play_services_ads_identifier',
-              'version': 'version:2@18.2.0.cr1',
-          },
-      ],
-      'condition': 'checkout_android',
-      'dep_type': 'cipd',
-  },
-  
   'src/third_party/android_deps/cipd/libs/com_google_android_gms_play_services_auth': {
-=======
-  'src/third_party/android_deps/libs/com_google_android_gms_play_services_auth': {
->>>>>>> 2e65a38ecd0 (Reduce minSdkVersion to 24 and Downgrade Google Ads Identifier Version (#6077))
       'packages': [
           {
               'package': 'chromium/third_party/android_deps/libs/com_google_android_gms_play_services_auth',
