--- cobalt/shell/BUILD.gn
+++ cobalt/shell/BUILD.gn
@@ -381,22 +381,14 @@
     "$root_gen_dir/base/tracing/protos/tracing_proto_resources.pak",
     "$root_gen_dir/components/ukm_resources.pak",
     "$root_gen_dir/cobalt/shell/cobalt_shell_resources.pak",
-<<<<<<< HEAD
-    "$root_gen_dir/content/attribution_internals_resources.pak",
-=======
->>>>>>> 2b97d9357b8 (Remove unneeded Chrome diagnostic tools (#6959))
     "$root_gen_dir/content/browser/resources/media/media_internals_resources.pak",
     "$root_gen_dir/content/browser/webrtc/resources/webrtc_internals_resources.pak",
     "$root_gen_dir/content/content_resources.pak",
     "$root_gen_dir/content/gpu_resources.pak",
     "$root_gen_dir/content/histograms_resources.pak",
-<<<<<<< HEAD
     "$root_gen_dir/content/private_aggregation_internals_resources.pak",
     "$root_gen_dir/content/process_resources.pak",
-    "$root_gen_dir/content/quota_internals_resources.pak",
     "$root_gen_dir/content/service_worker_resources.pak",
-=======
->>>>>>> 2b97d9357b8 (Remove unneeded Chrome diagnostic tools (#6959))
     "$root_gen_dir/content/test/web_ui_mojo_test_resources.pak",
     "$root_gen_dir/mojo/public/js/mojo_bindings_resources.pak",
     "$root_gen_dir/third_party/blink/public/resources/blink_resources.pak",
