--- gpu/config/skia_limits.cc
+++ gpu/config/skia_limits.cc
@@ -31,10 +31,11 @@
 #if !BUILDFLAG(IS_NACL)
 // The limit of the bytes allocated toward GPU resources in the GrContext's
 // GPU cache.
-MIRACLE_PARAMETER_FOR_INT(GetMaxLowEndGaneshResourceCacheBytes,
-                          kGrCacheLimitsFeature,
-                          "MaxLowEndGaneshResourceCacheBytes",
-                          48 * 1024 * 1024)
+[[maybe_unused]] MIRACLE_PARAMETER_FOR_INT(
+    GetMaxLowEndGaneshResourceCacheBytes,
+    kGrCacheLimitsFeature,
+    "MaxLowEndGaneshResourceCacheBytes",
+    48 * 1024 * 1024)
 
 MIRACLE_PARAMETER_FOR_INT(GetMaxHighEndGaneshResourceCacheBytes,
                           kGrCacheLimitsFeature,
@@ -91,36 +92,16 @@
 
 // We can't call AmountOfPhysicalMemory under NACL, so leave the default.
 #if !BUILDFLAG(IS_NACL)
-<<<<<<< HEAD
   if (base::SysInfo::IsLowEndDevice()) {
+#if BUILDFLAG(IS_COBALT)
+    *max_resource_cache_bytes = 0;
+#else
     *max_resource_cache_bytes = GetMaxLowEndGaneshResourceCacheBytes();
+#endif
     *max_glyph_cache_texture_bytes = GetMaxLowEndGlyphCacheTextureBytes();
   } else if (base::SysInfo::AmountOfPhysicalMemoryMB() >=
              GetHighEndMemoryThresholdMB()) {
     *max_resource_cache_bytes = GetMaxHighEndGaneshResourceCacheBytes();
-=======
-  // The limit of the bytes allocated toward GPU resources in the GrContext's
-  // GPU cache.
-#if !BUILDFLAG(IS_COBALT)
-  constexpr size_t kMaxLowEndGaneshResourceCacheBytes = 48 * 1024 * 1024;
-#endif
-  constexpr size_t kMaxHighEndGaneshResourceCacheBytes = 256 * 1024 * 1024;
-  // Limits for glyph cache textures.
-  constexpr size_t kMaxLowEndGlyphCacheTextureBytes = 1024 * 512 * 4;
-  // High-end / low-end memory cutoffs.
-  constexpr uint64_t kHighEndMemoryThreshold = 4096ULL * 1024 * 1024;
-
-  if (base::SysInfo::IsLowEndDevice()) {
-#if BUILDFLAG(IS_COBALT)
-    *max_resource_cache_bytes = 0;
-#else
-    *max_resource_cache_bytes = kMaxLowEndGaneshResourceCacheBytes;
-#endif
-    *max_glyph_cache_texture_bytes = kMaxLowEndGlyphCacheTextureBytes;
-  } else if (base::SysInfo::AmountOfPhysicalMemory() >=
-             kHighEndMemoryThreshold) {
-    *max_resource_cache_bytes = kMaxHighEndGaneshResourceCacheBytes;
->>>>>>> 0eeef7e51bc (Reduce Skia's GPU resource cache size to lower memory usage (#7531))
   }
 #endif
 }
