--- third_party/blink/renderer/platform/BUILD.gn
+++ third_party/blink/renderer/platform/BUILD.gn
@@ -2429,19 +2429,11 @@
     sources += [ "text/locale_icu_test.cc" ]
   }
 
-<<<<<<< HEAD
-  sources += [ "testing/run_all_tests.cc" ]
-=======
-  if (enable_av1_decoder) {
-    sources += [ "image-decoders/avif/avif_image_decoder_test.cc" ]
-  }
-
   if (is_cobalt_hermetic_build) {
     sources += [ "testing/starboard/run_all_tests.cc" ]
   } else {
     sources += [ "testing/run_all_tests.cc" ]
   }
->>>>>>> f9a9e5d581c (Add entry points for modular targets (#6302))
 
   configs += [
     ":blink_platform_pch",
