--- third_party/blink/renderer/bindings/idl_in_modules.gni
+++ third_party/blink/renderer/bindings/idl_in_modules.gni
@@ -1391,7 +1391,6 @@
   ]
 }
 
-<<<<<<< HEAD
 # Compute Pressure
 if (enable_compute_pressure) {
   static_idl_files_in_modules += [
@@ -1400,7 +1399,8 @@
     "//third_party/blink/renderer/modules/compute_pressure/pressure_record.idl",
     "//third_party/blink/renderer/modules/compute_pressure/pressure_update_callback.idl",
   ]
-=======
+}
+
 # SourceBufferWriteHead
 # An extension to the SourceBuffer interface that allows web apps to check the
 # highest presentation timestamp written to the Renderer.
@@ -1408,7 +1408,6 @@
   static_idl_files_in_modules += get_path_info(
           [ "//third_party/blink/renderer/modules/mediasource/source_buffer_write_head.idl" ],
           "abspath")
->>>>>>> 49e76a18382 (Implement SourceBuffer.writeHead attribute (#4559))
 }
 
 # Statically-defined (not runtime-generated) IDL files in 'modules' component.
