--- base/test/run_all_unittests.cc
+++ base/test/run_all_unittests.cc
@@ -13,16 +13,14 @@
 #include "base/win/com_init_util.h"
 #endif  // BUILDFLAG(IS_WIN)
 
-<<<<<<< HEAD
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
 #include "base/process/set_process_title_linux.h"
 #endif
-=======
+
 #if BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
 #include "base/test/allow_check_is_test_for_testing.h"
 #include "starboard/client_porting/wrap_main/wrap_main.h"
 #endif  // BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
->>>>>>> d0447f1e805 (Run base_unittests. (#5069))
 
 namespace base {
 
