--- media/mojo/services/mojo_demuxer_stream_adapter.cc
+++ media/mojo/services/mojo_demuxer_stream_adapter.cc
@@ -77,18 +77,13 @@
 void MojoDemuxerStreamAdapter::OnStreamReady(
     Type type,
     mojo::ScopedDataPipeConsumerHandle consumer_handle,
-<<<<<<< HEAD
     const std::optional<AudioDecoderConfig>& audio_config,
-    const std::optional<VideoDecoderConfig>& video_config) {
-=======
-    const absl::optional<AudioDecoderConfig>& audio_config,
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
-    const absl::optional<VideoDecoderConfig>& video_config,
+    const std::optional<VideoDecoderConfig>& video_config,
     const std::string& mime_type) {
 #else   // BUILDFLAG(USE_STARBOARD_MEDIA)
-    const absl::optional<VideoDecoderConfig>& video_config) {
+    const std::optional<VideoDecoderConfig>& video_config) {
 #endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
->>>>>>> 7a71b7745b8 ([media] Pass full mime type to SbPlayer via MojoRenderer (#6110))
   DVLOG(1) << __func__;
   DCHECK_EQ(UNKNOWN, type_);
   DCHECK(consumer_handle.is_valid());
