--- build/config/clang/clang.gni
+++ build/config/clang/clang.gni
@@ -17,18 +17,15 @@
 
   enable_check_raw_ptr_fields =
       build_with_chromium && !is_official_build &&
-<<<<<<< HEAD
       ((is_linux && !is_castos) || (is_android && !is_cast_android) || is_mac ||
-       is_win || is_chromeos_lacros || is_chromeos_ash)
+       is_win || is_chromeos_lacros || is_chromeos_ash) && !chrobalt
 
   # TODO(crbug.com/1446146): Merge with enable_check_raw_ptr_fields once both
   # checks are activated on the same set of platforms.
   enable_check_raw_ref_fields =
       build_with_chromium && !is_official_build &&
       ((is_linux && !is_castos) || (is_android && !is_cast_android) || is_win)
-=======
-      ((is_linux && !is_castos) || (is_android && !is_cast_android)) && !chrobalt
->>>>>>> 7521291e258 (Add starboard build to android (#4259))
+       && !chrobalt
 
   clang_base_path = default_clang_base_path
 
