--- base/allocator/partition_allocator/src/partition_alloc/partition_alloc_config.h
+++ base/allocator/partition_allocator/src/partition_alloc/partition_alloc_config.h
@@ -5,6 +5,7 @@
 #ifndef PARTITION_ALLOC_PARTITION_ALLOC_CONFIG_H_
 #define PARTITION_ALLOC_PARTITION_ALLOC_CONFIG_H_
 
+#include "build/build_config.h"
 #include "partition_alloc/build_config.h"
 #include "partition_alloc/buildflags.h"
 
@@ -172,11 +173,8 @@
 
 // On these platforms, lock all the partitions before fork(), and unlock after.
 // This may be required on more platforms in the future.
-<<<<<<< HEAD
-#define PA_CONFIG_HAS_ATFORK_HANDLER()                 \
-  (PA_BUILDFLAG(IS_APPLE) || PA_BUILDFLAG(IS_LINUX) || \
-   PA_BUILDFLAG(IS_CHROMEOS))
-
+#define PA_CONFIG_HAS_ATFORK_HANDLER()                                      \
+  (PA_BUILDFLAG(IS_APPLE) || (PA_BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_STARBOARD)) || PA_BUILDFLAG(IS_CHROMEOS))
 // Enable shadow metadata.
 //
 // With this flag, shadow pools will be mapped, on which writable shadow
@@ -189,10 +187,6 @@
 #else
 #define PA_CONFIG_ENABLE_SHADOW_METADATA() 0
 #endif
-=======
-#define PA_CONFIG_HAS_ATFORK_HANDLER()                                      \
-  (BUILDFLAG(IS_APPLE) || (BUILDFLAG(IS_LINUX) && !BUILDFLAG(IS_STARBOARD)) || BUILDFLAG(IS_CHROMEOS))
->>>>>>> 341f18f31b1 ([POSIX] Remove unnecessary pthread_atfork call (#5924))
 
 // PartitionAlloc uses PartitionRootEnumerator to acquire all
 // PartitionRoots at BeforeFork and to release at AfterFork.
