--- cobalt/shell/browser/shell.h
+++ cobalt/shell/browser/shell.h
@@ -144,21 +144,7 @@
   bool IsFullscreenForTabOrPending(const WebContents* web_contents) override;
   blink::mojom::DisplayMode GetDisplayMode(
       const WebContents* web_contents) override;
-<<<<<<< HEAD
-#if !BUILDFLAG(IS_ANDROID)
-  void RegisterProtocolHandler(RenderFrameHost* requesting_frame,
-                               const std::string& protocol,
-                               const GURL& url,
-                               bool user_gesture) override;
-  void UnregisterProtocolHandler(RenderFrameHost* requesting_frame,
-                                 const std::string& protocol,
-                                 const GURL& url,
-                                 bool user_gesture) override;
-#endif
   void RequestPointerLock(WebContents* web_contents,
-=======
-  void RequestToLockMouse(WebContents* web_contents,
->>>>>>> cf7a664ffd2 (Create cobalt_shell_test_lib static library (#7158))
                           bool user_gesture,
                           bool last_unlocked_by_target) override;
   void CloseContents(WebContents* source) override;
@@ -209,16 +195,14 @@
     delay_popup_contents_delegate_for_testing_ = delay;
   }
 
-<<<<<<< HEAD
   void set_hold_file_chooser() { hold_file_chooser_ = true; }
 
   // Counts both RunFileChooser and EnumerateDirectory.
   size_t run_file_chooser_count() const { return run_file_chooser_count_; }
-=======
+
  protected:
   // Finishes initialization of a new shell window.
   static void FinishShellInitialization(Shell* shell);
->>>>>>> cf7a664ffd2 (Create cobalt_shell_test_lib static library (#7158))
 
  private:
   class DevToolsWebContentsObserver;
