--- base/BUILD.gn
+++ base/BUILD.gn
@@ -3745,8 +3745,6 @@
     ]
   }
 
-<<<<<<< HEAD
-=======
   # TODO: b/402471389 - Cobalt: try to enable these tests.
   if (enable_cobalt_hermetic_hacks) {
     sources -= [
@@ -3756,35 +3754,6 @@
     ]
   }
 
-  if (is_mac) {
-    sources += [
-      "allocator/partition_allocator/shim/allocator_interception_mac_unittest.mm",
-      "allocator/partition_allocator/shim/malloc_zone_functions_mac_unittest.cc",
-      "enterprise_util_mac_unittest.mm",
-      "mac/backup_util_unittest.mm",
-      "mac/bind_objc_block_unittest.mm",
-      "mac/call_with_eh_frame_unittest.mm",
-      "mac/dispatch_source_mach_unittest.cc",
-      "mac/foundation_util_unittest.mm",
-      "mac/mac_util_unittest.mm",
-      "mac/mach_port_rendezvous_unittest.cc",
-      "mac/scoped_mach_vm_unittest.cc",
-      "mac/scoped_nsobject_unittest.mm",
-      "mac/scoped_objc_class_swizzler_unittest.mm",
-      "mac/scoped_sending_event_unittest.mm",
-      "message_loop/message_pump_mac_unittest.mm",
-      "power_monitor/thermal_state_observer_mac_unittest.mm",
-      "process/memory_unittest_mac.h",
-      "process/memory_unittest_mac.mm",
-      "strings/sys_string_conversions_mac_unittest.mm",
-    ]
-  }
-
-  if (is_apple && enable_mach_absolute_time_ticks) {
-    sources += [ "time/time_mac_unittest.mm" ]
-  }
-
->>>>>>> 8e464fdffdf (More fixes for base_unittest build. (#5019))
   if (is_posix) {
     sources += [
       "files/dir_reader_posix_unittest.cc",
