--- cobalt/shell/app/shell_main_delegate.cc
+++ cobalt/shell/app/shell_main_delegate.cc
@@ -39,15 +39,9 @@
 #include "cobalt/shell/common/shell_switches.h"
 #include "cobalt/shell/gpu/shell_content_gpu_client.h"
 #include "cobalt/shell/renderer/shell_content_renderer_client.h"
-<<<<<<< HEAD
-#if defined(RUN_BROWSER_TESTS)
-#include "cobalt/shell/utility/shell_content_utility_client.h"  // nogncheck
-#endif  // defined(RUN_BROWSER_TESTS)
 #if !BUILDFLAG(IS_ANDROIDTV)
 #include "components/crash/core/common/crash_key.h"
 #endif
-=======
->>>>>>> ce748a3700b (Separate switches for test and gate renderer test logics in non-official builds (#6881))
 #include "components/memory_system/initializer.h"
 #include "components/memory_system/parameters.h"
 #include "content/common/content_constants_internal.h"
