--- /var/www/html/CGP/conf/config.php	2019-08-07 09:02:26.935953434 -0700
+++ conf/config.php	2019-08-07 08:41:05.058259144 -0700
@@ -29,7 +29,7 @@
 #$CONFIG['overview_filter']['interface'] = array('ti' => 'eth0', 't' => 'if_octets');
 
 # default plugins time range
-$CONFIG['time_range']['default'] = 86400;
+$CONFIG['time_range']['default'] = 3600 * 2;
 $CONFIG['time_range']['uptime']  = 31536000;
 
 # show load averages and used memory on overview page
@@ -38,6 +38,7 @@
 $CONFIG['showtime'] = false;
 
 $CONFIG['term'] = array(
+	'30minute'=> 60 * 30,
 	'2hour'	 => 3600 * 2,
 	'8hour'	 => 3600 * 8,
 	'day'	 => 86400,
@@ -51,7 +52,7 @@
 $CONFIG['network_datasize'] = 'bytes';
 
 # "png", "svg", "canvas" or "hybrid" (canvas on detail page, png on the others) graphs
-$CONFIG['graph_type'] = 'png';
+$CONFIG['graph_type'] = 'hybrid';
 
 # For canvas graphs, use 'async' or 'sync' fetch method
 $CONFIG['rrd_fetch_method'] = 'sync';
