--- build/config/BUILD.gn
+++ build/config/BUILD.gn
@@ -211,22 +211,10 @@
     # linking can have run-time side effects, nothing should be listed here.
     libs = []
   } else if (is_ios) {
-<<<<<<< HEAD
     # Targets should choose to explicitly link frameworks they require. Since
     # linking can have run-time side effects, nothing should be listed here.
     libs = []
-  } else if (is_linux || is_chromeos) {
-=======
-    # The libraries listed here will be specified for both the target and the
-    # host. Only the common ones should be listed here.
-    frameworks = [
-      "CoreFoundation.framework",
-      "CoreGraphics.framework",
-      "CoreText.framework",
-      "Foundation.framework",
-    ]
   } else if (is_linux && !is_cobalt_hermetic_build || is_chromeos) {
->>>>>>> b0c7342c983 (Remove sysroot from linux hermetic builds (#5734))
     libs = [
       "dl",
       "pthread",
