--- media/mojo/common/mojo_decoder_buffer_converter.cc
+++ media/mojo/common/mojo_decoder_buffer_converter.cc
@@ -272,27 +272,12 @@
       continue;
     }
 
-<<<<<<< HEAD
     size_t actually_read_bytes = 0;
     MojoResult result = consumer_handle_->ReadData(
         MOJO_WRITE_DATA_FLAG_NONE,
         // We may be starting to read a new buffer (|bytes_read_| == 0), or
         // recovering from a previous partial read (|bytes_read_| > 0).
         buffer->writable_span().subspan(bytes_read_), actually_read_bytes);
-=======
-    // We may be starting to read a new buffer (|bytes_read_| == 0), or
-    // recovering from a previous partial read (|bytes_read_| > 0).
-    DCHECK_GT(buffer_size, bytes_read_);
-    size_t num_bytes = buffer_size - bytes_read_;
-
-#if BUILDFLAG(USE_STARBOARD_MEDIA)
-    MojoResult result = MOJO_RESULT_OK;
-#else // BUILDFLAG(USE_STARBOARD_MEDIA)
-    MojoResult result =
-        consumer_handle_->ReadData(buffer->writable_data() + bytes_read_,
-                                   &num_bytes, MOJO_WRITE_DATA_FLAG_NONE);
-#endif // BUILDFLAG(USE_STARBOARD_MEDIA)
->>>>>>> 156b8fdf9c5 ([media] Avoid extra allocation and copy for DecoderBuffer on renderer process (#5464))
 
     if (IsPipeReadWriteError(result)) {
       OnPipeError(result);
@@ -462,18 +447,13 @@
     bytes_to_write = bytes_to_write.subspan(bytes_written_);
     DCHECK_GT(bytes_to_write.size(), 0u);
 
-<<<<<<< HEAD
     size_t actually_written_bytes = 0;
-    MojoResult result = producer_handle_->WriteData(
-        bytes_to_write, MOJO_WRITE_DATA_FLAG_NONE, actually_written_bytes);
-=======
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
     MojoResult result = MOJO_RESULT_OK;
 #else // BUILDFLAG(USE_STARBOARD_MEDIA)
     MojoResult result = producer_handle_->WriteData(
-        buffer->data() + bytes_written_, &num_bytes, MOJO_WRITE_DATA_FLAG_NONE);
+        bytes_to_write, MOJO_WRITE_DATA_FLAG_NONE, actually_written_bytes);
 #endif // BUILDFLAG(USE_STARBOARD_MEDIA)
->>>>>>> 156b8fdf9c5 ([media] Avoid extra allocation and copy for DecoderBuffer on renderer process (#5464))
 
     if (IsPipeReadWriteError(result)) {
       OnPipeError(result);
