--- base/process/set_process_title.cc
+++ base/process/set_process_title.cc
@@ -44,11 +44,8 @@
 
 // TODO(jrg): Find out if setproctitle or equivalent is available on Android.
 #if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_APPLE) && !BUILDFLAG(IS_SOLARIS) && \
-<<<<<<< HEAD:base/process/set_process_title.cc
-    !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_NACL)
-=======
-    !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_STARBOARD)
->>>>>>> e58c2c869c8 (Remove program_invocation_short_name (#6065)):content/common/set_process_title.cc
+    !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_NACL) && \
+    !BUILDFLAG(IS_STARBOARD)
 
 void SetProcessTitleFromCommandLine(const char** main_argv) {
   // Build a single string which consists of all the arguments separated
