--- components/performance_manager/service_worker_context_adapter.cc
+++ components/performance_manager/service_worker_context_adapter.cc
@@ -90,15 +90,11 @@
 
 void ServiceWorkerContextAdapter::RunningServiceWorker::
     RenderProcessHostDestroyed(content::RenderProcessHost* host) {
-<<<<<<< HEAD
-  NOTREACHED_NORETURN();
-=======
   // In --single-process mode, `RenderProcessExited` is never called as there is
   // no render process. The worker nodes still need to be cleaned up before the
   // process node.
   CHECK(content::RenderProcessHost::run_renderer_in_process());
   adapter_->OnRenderProcessExited(version_id_);
->>>>>>> c4417b5173b (Fix handling of RenderProcessHost destruction in single-process mode (#4894))
 }
 
 // ServiceWorkerContextAdapter::RunningServiceWorker ---------------------------
