--- content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java
+++ content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java
@@ -76,17 +76,10 @@
         mWindowAndroid.setAnimationPlaceholderView(
                 mShellManager.getContentViewRenderView().getSurfaceView());
 
-<<<<<<< HEAD
-        mStartupUrl = getUrlFromIntent(getIntent());
-=======
         // TODO(cobalt, b/376148547): set Chrobalt initial url and remove this function.
         if (mStartupUrl.isEmpty()) {
             mStartupUrl = getUrlFromIntent(getIntent());
         }
-        if (!TextUtils.isEmpty(mStartupUrl)) {
-            mShellManager.setStartupUrl(Shell.sanitizeUrl(mStartupUrl));
-        }
->>>>>>> 37be7d9f3b4 (`cobalt_apk` to build dev.cobalt.coat (#4304))
 
         if (CommandLine.getInstance().hasSwitch(RUN_WEB_TESTS_SWITCH)) {
             BrowserStartupController.getInstance()
