--- starboard/build/platform_path.gni
+++ starboard/build/platform_path.gni
@@ -18,7 +18,6 @@
 
 if (starboard_path == "") {
   declare_args() {
-<<<<<<< HEAD
     starboard_target_platform = ""
   }
 
@@ -30,41 +29,20 @@
         starboard_target_platform = "android-arm64"
       } else if (target_cpu == "x86") {
         starboard_target_platform = "android-x86"
-=======
-    target_platform = ""
-  }
-
-  if (target_platform == "") {
-    if (target_os == "android") {
-      if (target_cpu == "arm") {
-        target_platform = "android-arm"
-      } else if (target_cpu == "arm64") {
-        target_platform = "android-arm64"
-      } else if (target_cpu == "x86") {
-        target_platform = "android-x86"
->>>>>>> b80977bcd89 (Add starboard build to android (#4259))
       } else {
         assert(false, "Unsupported android target cpu ${target_cpu}.")
       }
     } else if (target_os == "linux") {
       if (target_cpu == "x64") {
-<<<<<<< HEAD
         starboard_target_platform = "linux-x64x11"
       } else if (target_cpu == "arm") {
         starboard_target_platform = "raspi-2"
-=======
-        target_platform = "linux-x64x11"
->>>>>>> b80977bcd89 (Add starboard build to android (#4259))
       } else {
         assert(false, "Unsupported linux target cpu ${target_cpu}.")
       }
     } else {
       # Default to linux-x64x11 for chromium toolchains.
-<<<<<<< HEAD
       starboard_target_platform = "linux-x64x11"
-=======
-      target_platform = "linux-x64x11"
->>>>>>> b80977bcd89 (Add starboard build to android (#4259))
     }
   }
 
