--- build/config/clang/clang.gni
+++ build/config/clang/clang.gni
@@ -18,7 +18,6 @@
 
   enable_check_raw_ptr_fields =
       build_with_chromium && !is_official_build &&
-<<<<<<< HEAD
       ((is_linux && !is_castos) || (is_android && !is_cast_android) || is_mac ||
        is_win || is_chromeos_lacros || is_chromeos_ash) && !is_cobalt
 
@@ -28,9 +27,6 @@
       build_with_chromium && !is_official_build &&
       ((is_linux && !is_castos) || (is_android && !is_cast_android) || is_win)
        && !is_cobalt
-=======
-      ((is_linux && !is_castos) || (is_android && !is_cast_android))
->>>>>>> b93382f1c611 (Set certain upstream GN flags correctly in args.gn (#4419))
 
   clang_base_path = default_clang_base_path
 
