--- BUILD.gn
+++ BUILD.gn
@@ -831,17 +831,10 @@
       deps += [ "//components/cronet:cronet_tests" ]
     }
   }
-<<<<<<< HEAD
 
   if (is_chrome_for_testing) {
     deps += [ "//chrome/browser/chrome_for_testing:chrome_for_testing" ]
   }
-
-  if (chrobalt) {
-    deps += [ "//chrobalt:gn_all" ]
-  }
-=======
->>>>>>> 62c20771c03 (Revert "Build starboard shared library for linux-x64x11 (#4215)" (#4245))
 }
 
 # All Rust targets. This is provided for convenience while developing Rust to
