--- base/allocator/partition_allocator/src/partition_alloc/partition_alloc_config.h
+++ base/allocator/partition_allocator/src/partition_alloc/partition_alloc_config.h
@@ -56,13 +56,8 @@
 #define PA_CONFIG_DYNAMICALLY_SELECT_POOL_SIZE() 0
 #endif  // PA_BUILDFLAG(HAS_64_BIT_POINTERS) && BUILDFLAG(IS_IOS)
 
-<<<<<<< HEAD
 #if PA_BUILDFLAG(HAS_64_BIT_POINTERS) && \
-    (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_ANDROID))
-=======
-#if BUILDFLAG(HAS_64_BIT_POINTERS) && \
     (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_ANDROID)) && !defined(IS_COBALT_HERMETIC_BUILD)
->>>>>>> 08381675a91 (Build nplb hermetically (#4587))
 #include <linux/version.h>
 // TODO(bikineev): Enable for ChromeOS.
 #define PA_CONFIG_STARSCAN_UFFD_WRITE_PROTECTOR_SUPPORTED() \
@@ -100,8 +95,12 @@
 
 // POSIX is not only UNIX, e.g. macOS and other OSes. We do use Linux-specific
 // features such as futex(2).
+#if defined(IS_COBALT_HERMETIC_BUILD)
+#define PA_CONFIG_HAS_LINUX_KERNEL() false
+#else
 #define PA_CONFIG_HAS_LINUX_KERNEL() \
-  (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID)) && !defined(IS_COBALT_HERMETIC_BUILD)
+  (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID))
+#endif
 
 // On some platforms, we implement locking by spinning in userspace, then going
 // into the kernel only if there is contention. This requires platform support,
