--- mojo/core/message_pipe_unittest.cc
+++ mojo/core/message_pipe_unittest.cc
@@ -375,13 +375,8 @@
     MojoClose(p[i]);
 }
 
-<<<<<<< HEAD
-#if BUILDFLAG(IS_IOS)
-// TODO(crbug.com/40257752): Test currently fails on iOS.
-=======
 #if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_STARBOARD)
-// TODO(crbug.com/1418597): Test currently fails on iOS.
->>>>>>> 5570c14d97c ([POSIX] Disable multi-process functions (#6146))
+// TODO(crbug.com/40257752): Test currently fails on iOS.
 #define MAYBE_SharedBufferHandlePingPong DISABLED_SharedBufferHandlePingPong
 #else
 #define MAYBE_SharedBufferHandlePingPong SharedBufferHandlePingPong
