--- base/task/sequenced_task_runner.cc
+++ base/task/sequenced_task_runner.cc
@@ -86,9 +86,6 @@
 // static
 const scoped_refptr<SequencedTaskRunner>&
 SequencedTaskRunner::GetCurrentDefault() {
-<<<<<<< HEAD
-  CHECK(HasCurrentDefault())
-=======
 #if BUILDFLAG(IS_COBALT)
   if (current_default_handle) {
     return current_default_handle->task_runner_;
@@ -97,8 +94,7 @@
     new scoped_refptr<SequencedTaskRunner>();
   return *null_sequenced_task_runner;
 #endif
-  CHECK(current_default_handle)
->>>>>>> 773a674f23a (base/task + base/threading: Remove RecordQueuingDelayedTaskMetrics and TLS lookups (#6212))
+  CHECK(HasCurrentDefault())
       << "Error: This caller requires a sequenced context (i.e. the current "
          "task needs to run from a SequencedTaskRunner). If you're in a test "
          "refer to //docs/threading_and_tasks_testing.md.";
