--- /usr/local/google/home/kaidokert/code/2ndproject/m120/3e92121/conflict/content/shell/browser/shell_content_browser_client.cc	2025-06-16 13:45:41.604198109 -0700
+++ /usr/local/google/home/kaidokert/code/2ndproject/m120/3e92121/resolved/content/shell/browser/shell_content_browser_client.cc	2025-06-16 13:48:41.561164412 -0700
@@ -299,10 +299,10 @@
   return pref_service_factory.Create(pref_registry);
 }
 
-<<<<<<< HEAD
 bool AreIsolatedWebAppsEnabled() {
   return base::FeatureList::IsEnabled(features::kIsolatedWebApps);
-=======
+}
+
 }  // namespace
 
 std::string GetShellUserAgent() {
@@ -318,11 +318,8 @@
 
   return GetShellFullUserAgent();
 #endif
->>>>>>> 3e92121a24891 (`cobalt_apk` to build dev.cobalt.coat (#4304))
 }
 
-}  // namespace
-
 std::string GetShellLanguage() {
   return "en-us,en";
 }
