--- content/app/content_main.cc
+++ content/app/content_main.cc
@@ -333,15 +333,7 @@
     CommonSubprocessInit();
   exit_code = content_main_runner->Run();
 
-<<<<<<< HEAD
-#if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_IOS)
-=======
-#if BUILDFLAG(IS_MAC)
-  autorelease_pool.reset();
-#endif
-
 #if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_IOS) && !BUILDFLAG(IS_STARBOARD)
->>>>>>> 8da63bba955 (Use the Starboard message pump. (#4772))
   content_main_runner->Shutdown();
 #endif
 
