--- base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_internals.h
+++ base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_internals.h
@@ -9,11 +9,7 @@
 
 #if defined(__GNUC__)
 
-<<<<<<< HEAD
 #if PA_BUILDFLAG(IS_POSIX) && !defined(IS_COBALT_HERMETIC_BUILD)
-=======
-#if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_COBALT_HERMETIC_BUILD)
->>>>>>> a019a910af7 (Add BUILDFLAG IS_COBALT_HERMETIC_BUILD (#4966))
 #include <sys/cdefs.h>  // for __THROW
 #endif
 
