--- BUILD.gn
+++ BUILD.gn
@@ -871,7 +871,6 @@
       deps += [ "//components/cronet:cronet_tests" ]
     }
   }
-<<<<<<< HEAD
 
   if (is_chrome_for_testing) {
     deps += [ "//chrome/browser/chrome_for_testing:chrome_for_testing" ]
@@ -880,15 +879,9 @@
   if (checkout_src_internal) {
     deps += [ "//internal:all" ]
   }
-  if (chrobalt) {
-    deps += [ "//chrobalt:gn_all" ]
-  }
-  if (chrobalt) {
-    deps += [ "//chrobalt:gn_all" ]
-=======
+  
   if (is_cobalt) {
     deps += [ "//cobalt:gn_all" ]
->>>>>>> 55799aec18e (Rename Chrobalt to Cobalt. (#4276))
   }
 }
 
