--- build/config/BUILDCONFIG.gn
+++ build/config/BUILDCONFIG.gn
@@ -179,16 +179,15 @@
 assert(!(current_os == "ios" && is_component_build),
        "Can't use component build on iOS")
 
-<<<<<<< HEAD
 declare_args() {
   # Unsafe buffers. Location of file used by plugins to track portions of
   # the codebase which have been made manifestly safe.
   clang_unsafe_buffers_paths = ""
-=======
+}
+
 if (is_cobalt) {
 assert(!cobalt_is_release_build || is_official_build,
        "Release builds are only available in official builds.")
->>>>>>> fe2633555484 (Infer UA build_configuration from build config (#4484))
 }
 
 # ==============================================================================
