--- base/logging.cc
+++ base/logging.cc
@@ -483,33 +483,7 @@
 }
 #endif  // BUILDFLAG(IS_STARBOARD)
 
-<<<<<<< HEAD
-=======
-#if BUILDFLAG(IS_FUCHSIA)
-inline FuchsiaLogSeverity LogSeverityToFuchsiaLogSeverity(
-    LogSeverity severity) {
-  switch (severity) {
-    case LOGGING_INFO:
-      return FUCHSIA_LOG_INFO;
-    case LOGGING_WARNING:
-      return FUCHSIA_LOG_WARNING;
-    case LOGGING_ERROR:
-      return FUCHSIA_LOG_ERROR;
-    case LOGGING_FATAL:
-      // Don't use FX_LOG_FATAL, otherwise fx_logger_log() will abort().
-      return FUCHSIA_LOG_ERROR;
-  }
-  if (severity > -3) {
-    // LOGGING_VERBOSE levels 1 and 2.
-    return FUCHSIA_LOG_DEBUG;
-  }
-  // LOGGING_VERBOSE levels 3 and higher, or incorrect levels.
-  return FUCHSIA_LOG_TRACE;
-}
-#endif  // BUILDFLAG(IS_FUCHSIA)
-
 #if !BUILDFLAG(IS_STARBOARD)
->>>>>>> ef14ff42abe (Fix clang warnings for modular builds (#6138))
 void WriteToFd(int fd, const char* data, size_t length) {
   size_t bytes_written = 0;
   long rv;
