--- media/base/stream_parser_buffer.cc
+++ media/base/stream_parser_buffer.cc
@@ -77,19 +77,15 @@
                                        bool is_key_frame,
                                        Type type,
                                        TrackId track_id)
-<<<<<<< HEAD
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+    : DecoderBuffer(type, data, data_size),
+#else // BUILDFLAG(USE_STARBOARD_MEDIA)
     : DecoderBuffer(
           // TODO(crbug.com/40284755): Convert `StreamBufferParser` to
           // `size_t` and `base::span`.
           UNSAFE_BUFFERS(
               base::span(data, base::checked_cast<size_t>(data_size)))),
-=======
-#if BUILDFLAG(USE_STARBOARD_MEDIA)
-    : DecoderBuffer(type, data, data_size),
-#else // BUILDFLAG(USE_STARBOARD_MEDIA)
-    : DecoderBuffer(data, data_size),
 #endif // BUILDFLAG(USE_STARBOARD_MEDIA)
->>>>>>> 1776bebbd06 ([media] Improve DecoderBufferAllocator logging (#5036))
       decode_timestamp_(kNoDecodeTimestamp),
       config_id_(kInvalidConfigId),
       type_(type),
