--- content/shell/browser/shell_content_browser_client.cc
+++ content/shell/browser/shell_content_browser_client.cc
@@ -306,10 +306,6 @@
 #if defined(IS_COBALT)
 std::string GetShellUserAgent() {
   // TODO(cobalt b/376147838): Revert the changes here and instead rely on coablt_content_browser_client.cc for UA.
-<<<<<<< HEAD
-=======
-#if BUILDFLAG(IS_COBALT)
->>>>>>> 0528637c55f (Fix Cobalt condition for UA (#4347))
   return "Mozilla/5.0 (LINUX) Cobalt/26.lts.10.0000001-gold (unlike Gecko) v8/8.8.278.8-jit gles Evergreen/4.10.2 Evergreen-Full Evergreen-Uncompressed Starboard/15, odm_TV_chipset_2024/fw-01-23.45 (brand, model)";
 }
 #endif
