--- ui/gl/BUILD.gn
+++ ui/gl/BUILD.gn
@@ -197,21 +197,13 @@
   }
 
   if (use_ozone) {
-<<<<<<< HEAD
-    if (!is_fuchsia) {
+    if (!is_fuchsia && !is_cobalt_hermetic_build) {
       if (!use_static_angle) {
         data_deps += [
           "//third_party/angle:libEGL",
           "//third_party/angle:libGLESv2",
         ]
       }
-=======
-    if (!is_fuchsia && !is_cobalt_hermetic_build) {
-      data_deps += [
-        "//third_party/angle:libEGL",
-        "//third_party/angle:libGLESv2",
-      ]
->>>>>>> 164d14e440c (Remove sysroot from linux hermetic builds (#5734))
       if (angle_shared_libvulkan && !is_chromeos) {
         data_deps += [ "//third_party/vulkan-loader/src:libvulkan" ]
       }
